<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Welcome to the UKRI Digital Research Skills Catalyst</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" integrity="sha384-ZvpUoO/+PpLXR1lu4jmpXWu80pZlYUAfxl5NsBMWOEPSjUn/6Z/hRTt8+pR6L4N2" crossorigin="anonymous"></script><script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-587c61ba64f3a5504c4d52d930310e48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-4690a3ddcde68fcbfe3727ff945e31b1.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script type="module" src="site_libs/quarto-ojs/quarto-ojs-runtime.js"></script>
<link href="site_libs/quarto-ojs/quarto-ojs.css" rel="stylesheet">
<script src="https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.min.js" integrity="sha384-c9c+LnTbwQ3aujuU7ULEPVvgLs+Fn6fJUvIGTsuu1ZcCf11fiEubah0ttpca4ntM sha384-6V1/AdqZRWk1KAlWbKBlGhN7VG4iE/yAZcO6NZPMF8od0vukrvr0tg4qY6NSrItx" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


<link rel="stylesheet" href="style.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./assets/ukri-drsc-logo-transparent.png" alt="Logo for the UKRI Digital Research Skills Catalyst" class="navbar-logo light-content">
    <img src="./assets/ukri-drsc-logo-transparent.png" alt="Logo for the UKRI Digital Research Skills Catalyst" class="navbar-logo dark-content">
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./consult.html"> 
<span class="menu-text">Consult an Expert</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./news.html"> 
<span class="menu-text">News &amp; Events</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content column-page" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Welcome to the UKRI Digital Research Skills Catalyst</h1>
<p class="subtitle lead">A national hub for researcher and innovator development.</p>
</div>



<div class="quarto-title-meta column-page">

    
  
    
  </div>
  


</header>


<p>Boost your digital research skills, accelerate your research impactâ€”discover 80+ learning resources, events, and expert-led training in one central hub with the UKRI Digital Research Skills Catalyst. Find out more <a href="./about.html">here</a>.</p>
<div id="20fec11b" class="cell" data-cache="true" data-execution_count="1">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This cell loads the CSV data directly from the Google sheet via its published CSV-specific URL.</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># We need to load in Python, as OJS doesn't have a caching system.</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Once loaded, we "publish" the variable globally to OJS.</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>course_data_cached <span class="op">=</span> pd.read_csv(<span class="st">"https://docs.google.com/spreadsheets/d/e/2PACX-1vRSrvsfFfVwokza_WP9JIzd4Wfg6OKPBJcwelLTqYn1SgigZXnfcU6_apN5gWTMF79n4CRQFNOJ5w6M/pub?gid=1012757406&amp;single=true&amp;output=csv"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>ojs_define(course_data_cached <span class="op">=</span> course_data_cached)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<script type="ojs-define">{"contents": [{"name": "course_data_cached", "value": {"headline": ["Metagenomics", "Metagenomics for Environmental Scientists", "Prenomics", "Core R", "Create Your Own AWS Instance", "Genomics", "Statistically Useful Experimental Design", "Image Analysis for Microscopy", "Introduction to Image Analysis for Plant Phenotyping", "Introduction to Python for Health Research", "Making clinical datasets FAIR", "Sequence data submission to ENA", "FAIR data management solutions", "Recommended Metadata for Biological Images", "Filter, plot and explore single-cell RNA-seq data", "FAIR Bioimage Metadata", "RNA-req Data readiness for machine learning applications", "FAIR Galaxy Training Material", "FAIR in a nutshell", "FAIR data management for Long-term agriculture experiments", "FAIR pointers", "Workflows with Nextflow", "Workflows with Snakemake", "FAIR in Biological Practice", "High Dimentional Statistics in R", "Introduction to Conda", "Introduction to Machine Learning In Python", "Introduction to Statistics with R", "Python Fundamentals", "Data Handling", "Network Science", "Machine Learning", "Artificial Intelligence", "Introduction to Plant Phenotyping Technologies", "Machine Learning for Image Data", "Experimental Design for Machine Learning", "Deep Learning for Bioscientists", "Generative AI for Plant Phenotyping", "Artificial Intelligence in Health Analytics", "Introduction to R for Health Research", "Unix and IT Infrastructure for Data Analysis", "Machine Learning for Health Research", "Prediction Modelling for Health Research", "Demystifying AI", "Using Spreadsheets to Record Data and Metadata", "Intermediate Python and Software Engineering", "NGS: Introduction and Pre-processing Pipeline", "Basic R with Data Carpentry", "Natural Language Processing for Health Research: Part 1", "Introduction to Statistics", "Statistics with R", "NGS: Single-cell RNA-sequencing Analysis", "Applied Artificial Intelligence for Health Research", "NGS on the Galaxy Platform", "NGS: RNA-seq", "NGS: ATAC-seq", "Sequence Alignment and BLAST"], "accessibilityAPI": [null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null], "accessMode": ["textual", "textual", "textual", "textual", "textual", "textual", "textual", "textual", "visual", "visual", "textual,  visual", "textual", "textual", "textual,  visual", "textual,  visual", "textual", "textual", "textual", "textual", "textual", "textual", "textual", "textual", "textual", "textual", "textual", "textual, auditory", "textual", "textual, visual", "textual, visual", "textual, visual", "textual, visual", "textual, visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual", "visual"], "audience": ["Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Postgraduate student, Postgraduate researcher, Bioinformatician", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Engineer, Trainer, Industry, Public sector, Research Software Engineer", "Postgraduate researcher, Researcher, Clinician, Industry", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Research Software Engineer, Bioinformatician, Research skills educator, Undergraduate student", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Trainer, Research Software Engineer, Bioinformatician, Open Research Specialist, Undergraduate student, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Engineer, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Engineer, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Engineer, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator, Undergraduate student", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Engineer, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Industry, Bioinformatician, Public sector, Open Research Specialist", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Research Software Engineer, Bioinformatician, Open Research Specialist, Industry, Public sector", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Open Research Specialist, Bioinformatician, Research Software Engineer, Undergraduate student, Public sector, Industry", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Industry, Research Software Engineer, Bioinformatician, Open Research Specialist, Public sector", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Industry, Research Software Engineer, Bioinformatician, Open Research Specialist, Public sector", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Industry, Public sector, Research Software Engineer, Bioinformatician, Open Research Specialist", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Clinician, Research Software Engineer, Bioinformatician, Open Research Specialist, Research skills educator, Industry", "Postgraduate researcher, Postgraduate student, Researcher, Clinician, Research Software Engineer, Bioinformatician, Research skills educator", "Postgraduate researcher, Postgraduate student, Researcher, Clinician, Research Software Engineer, Bioinformatician, Research skills educator", "Postgraduate researcher, Postgraduate student, Researcher, Clinician, Research Software Engineer, Bioinformatician, Research skills educator", "Postgraduate researcher, Postgraduate student, Researcher, Clinician, Research Software Engineer, Bioinformatician, Research skills educator", "Postgraduate researcher, Postgraduate student, Researcher, Clinician, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Industry, Public sector, Research Software Engineer, Research skills educator, Trainer, Engineer", "Undergraduate student, Postgraduate student, Postgraduate researcher, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Trainer, Engineer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Trainer, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Undergraduate student, Postgraduate student, Postgraduate researcher, Researcher, Trainer, Industry, Public sector, Research Software Engineer, Bioinformatician, Research skills educator", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Industry", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Industry, Public sector", "Postgraduate student, Postgraduate researcher, Researcher, Bioinformatician, Research Software Engineer", "Postgraduate student, Postgraduate researcher, Researcher, Clinician, Public sector", "Postgraduate student, Researcher, Public sector", "Postgraduate student, Researcher, Clinician, Public sector, Postgraduate researcher", "Bioinformatician, Public sector, Undergraduate student, Postgraduate student, Researcher, Clinician, Engineer, Trainer, Research skills educator, Postgraduate researcher", "Postgraduate student, Researcher, Clinician, Engineer, Research Software Engineer, Public sector, Postgraduate researcher", "Bioinformatician, Postgraduate student, Researcher, Postgraduate researcher", "Postgraduate student, Researcher, Bioinformatician", "Postgraduate student, Researcher, Clinician, Public sector, Postgraduate researcher", "Postgraduate student, Researcher, Postgraduate researcher", "Researcher, Postgraduate student, Clinician, Bioinformatician", "Researcher, Postgraduate student, Clinician, Bioinformatician, Postgraduate researcher", "Researcher, Postgraduate student, Clinician, Bioinformatician, Public sector, Research Software Engineer", "Researcher, Postgraduate student, Clinician, Bioinformatician, Postgraduate researcher", "Researcher, Postgraduate student, Clinician, Bioinformatician", "Researcher, Postgraduate student, Clinician, Bioinformatician, Postgraduate researcher", "Researcher, Postgraduate student, Clinician, Bioinformatician, Postgraduate researcher"], "author": ["Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E", "Rand, E", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E", "Health+BioscienceIDEAS", "French, Andrew; Pridmore, Tony; Mellor, Nathan", "Roberts, Angus; Stahl, Daniel; Abdullah, Zahra; Langham, John;", "Sandra Ng, Tim Beck, Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Sonal Henson, Munazah Andrabi, Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Laura Cooper, Julia Jakiela, Katarzyna Kamieniecka, Krzysztof Poterlowicz", "Julia Jakiela, Wendi Bacon ", "Laura Coope, Julia Jakiela, Katarzyna Kamieniecka, Krzysztof Poterlowicz", "Edward Parkinson, Munazah Andrabi, Katarzyna Kamieniecka", "\nKatarzyna Kamieniecka, Krzysztof Poterlowicz", "\nKatarzyna Kamieniecka, Krzysztof Poterlowicz", "Richard Ostler", "Robert Andrews, Nick Juty, Munazah Andrabi, Nicola Soranzo, Sara Morsy, Kellie Snow, Korneel Hens, Philippe Rocca-Serra, Laura Cooper, Xenia Perez Sitja, Andrew Mason, Branka Franicevic, Saskia Lawson-Tovey, Krzysztof Poterlowicz, Katarzyna Kamieniecka", "Grimes, G. R., Floden, E., Di Tommaso, P., Ewels, P., & Garcia, M.", "Booth, T., Medd, N., Dunn-Davies, H., Emelianova, K., & Turner, F.", "Zieli\u0144ski, T. & Romanowski, A.", "O'Callaghan, A., Ewing, A., & Vallejos, C.", "Pugh, D. R.", "Pollard, Tom; Peru, Giacomo", "Herman, E., Rand, E., & Stewart, A.", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Atkinson, Jonathan; Postnikov, Andrey; Gaju, Oorbessy", "French, Andrew; Parsons, Simon", "Pound, Michael; French, Andrew; Mellor, Nathan", "Pound, Michael; Mellor, Nathan", "Pound, Michael; French, Andrew; Mellor, Nathan", "Cummins, Nicholas; Dinely, Judith; Stahl, Daniel; Abdullah, Zahra; Langham, John;", "Carr, Ewan; Ginestet, Cedric; Abdullah, Zahra; Langham, John; Stahl, Daniel;", "Flaviani, Flavia; Vigilante, Alessandra", "Ibrahim, Zina; Stahl, Daniel; Abdullah, Zahra; Langham, John; Qian, Linglong; Lehchevska, Nicole;", "Iniesta, Raquel; Agbedjro, Deborah; Shamsutdinova, Diana; Ginestet, Cedric; Abdullah, Zahra; Langham, John; Stahl, Daniel", "Cardoso, Jorge; Robinson, Emma; King, Andrew; Young, Alistair; Booth, Thomas; Da Silva, Mariana;", null, "Modat, Mark; Kerfoot, Eric; Barfoot, Theodore; Prokopenko, Denis; Cardoso, Jorge", "Hamrud, Eva; Thiery, Alex;", "Vigilante, Alessandra; Belokhvostova, Dasha; Cain, James;", "Roberts, Angus; Velupillai, Sumithra;", "Iniesta, Raquel; Dineley, Judith; Abdullah, Zahra; Vitoratou, Silia; Stahl, Daniel; Lehchevska, Nicole;", "Vigilante, Alessandra; Belokhvostova, Dasha; Cain, James;", "Hamid, Fursham; Lozano Gonzalez, Karla", "Cardoso, Jorge; Robinson, Emma; Da Silva, Mariana; Borges, Pedro; Langham, John; Crawford, Andre;", "Vigilante, Alessandra;", "Vigilante, Alessandra; Hamrud, Eva; Thiery, Alex;", "Hamrud, Eva; Thiery, Alex;", "Vigilante, Alessandra;"], "bespoke": ["yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no"], "competencyRequired": ["Bioscience: Understanding of what a microbiome is, Classical statistics: Understanding independent/expanatory variables and dependent/response variables, Bioscience: Understanding of a what genome is", "Bioscience: Understanding of what a microbiome is, Classical statistics: Understanding independent/expanatory variables and dependent/response variables, Bioscience: Understanding of a what genome is", null, null, "Command line: Ability to open a shell such as bash, Command line: Ability to login to remote systems from the command line, High Performance Computing: Ability to utilize cloud computing resources for data analysis", "Command line: Ability to open a shell such as bash, Command line: Ability to navigate filesystems from the command line, High Performance Computing: Ability to manage cloud-based computing environments, High Performance Computing: Ability to utilize cloud computing resources for data analysis, Computing: Understanding of hierarchical filesystems including root and home directories, \"Computing: Understanding working directories, relative and absolute paths\", Computing: Understanding of File types and formats including plain text files and the relation between file types and file extensions, \"Command line: Ability to move, delete, create and copy files and directories from command line\", Command line: Ability to use common command line functions, Command line: Ability to specify inputs and outputs for software from command line, Command line: Ability to login to remote systems from the command line, Command line: Ability to move files between local and remote systems from the command line, Command line: Ability to create and run shell scripts, Data management: Ability to document and share data analysis workflows", null, null, null, null, null, "Bioinformatics: Understanding of genomic data analysis, Bioscience: Understanding of DNA and genes, Bioscience: Understanding of a what genome is, Data management: Understanding of data integrity and quality control in bioinformatics", "Data management: Understanding of data integrity and quality control in bioinformatics, FAIR and Open Science: Understanding of Open Science principles", "FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, Image processing: Understanding of basics of image analysis", "Bioscience: Understanding of DNA and genes, Bioscience: Understanding of mutation, SNPs, and indels, Bioinformatics: Understanding of genomic data analysis, Classical statistics: Understanding independent/expanatory variables and dependent/response variables, Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous, Classical statistics: Understanding that variables have distributions", "Image processing: Understanding of basics of image analysis, Bioscience: Plant phenotyping", "R programming: Ability to create and use dataframes, R programming: Ability to plot graphs and figures using ggplot2, R programming: Proficiency in data manipulation and visualization, Classical statistics: Understanding independent/expanatory variables and dependent/response variables, Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous", "FAIR and Open Science: Understanding of FAIR principles, Bioinformatics: Awareness of the concept of bioinformatics pipelines", "FAIR and Open Science: Understanding of Open Science principles", "Bioscience: Plant phenotyping, Data management: Understanding of data integrity and quality control in bioinformatics", "FAIR and Open Science: Understanding of FAIR principles", "Command line: Ability to navigate filesystems from the command line, Computing: Understanding of pipelines and automation principles", "Command line: Ability to navigate filesystems from the command line, Command line: Ability to use common command line functions, Computing: Understanding of pipelines and automation principles", null, "R programming: Ability to create and use dataframes, R programming: Ability to plot graphs and figures using ggplot2, Classical statistics: Understanding independent/expanatory variables and dependent/response variables, \"Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous\"", "Command line: Ability to use common command line functions, Python Programming: Ability to use virtual environments and install packages using pip", "\"Python Programming: Ability to use variables and basic data types such as integers, floats, strings and booleans\", Python Programming: Ability to work with Python libraries for scientific computing such as NumPy and pandas, Python Programming: Ability to create and manipulate pandas DataFrames, Machine Learning: Ability to train and evaluate classification models using scikit-learn, Machine Learning: Understanding ML methodologies", "R programming: Ability to create and use dataframes, R programming: Ability to plot graphs and figures using ggplot2", "Computing: Understanding of hierarchical filesystems including root and home directories, Computing: Understanding of File types and formats including plain text files and the relation between file types and file extensions", "Computing: Ability to use an integrated development environment (IDE) or notebook environment such as Jupyter, \"Computing: Understanding of the difference between scripts, notebooks and interactive shells\", \"Python Programming: Ability to use variables and basic data types such as integers, floats, strings and booleans\", \"Python Programming: Ability to use control flow including if, elif, else statements\", Python Programming: Ability to use loops such as for and while to process data, Python Programming: Ability to define and call functions with arguments and return values, \"Python Programming: Ability to work with built-in data structures such as lists, tuples and dictionaries\", Python Programming: Ability to read from and write to files in Python, Python Programming: Ability to handle basic errors and exceptions in Python code, Python Programming: Ability to write simple Python scripts to automate repetitive tasks, Python Programming: Ability to use virtual environments and install packages using pip", "Classical statistics: Understanding independent/expanatory variables and dependent/response variables, \"Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous\", Classical statistics: Understanding that variables have distributions, Python Programming: Ability to work with Python libraries for scientific computing such as NumPy and pandas, Python Programming: Ability to use NumPy arrays for numerical computing, Python Programming: Ability to create and manipulate pandas DataFrames, Python Programming: Ability to import tabular data from CSV and Excel files into pandas, Python Programming: Ability to clean data including handling missing values and inconsistent entries, \"Python Programming: Ability to filter, sort and subset tabular data using pandas\", Python Programming: Ability to group and aggregate data in pandas, Python Programming: Ability to merge and join multiple DataFrames, \"Python Programming: Ability to calculate descriptive statistics such as mean, median, variance and quantiles in Python\", \"Python Programming: Ability to create basic data visualisations including histograms, boxplots, scatter plots and line plots\", Python Programming: Ability to use Matplotlib or similar libraries for publication-quality plots, Data management: Ability to document and share data analysis workflows, Data management: Understanding of data integrity and quality control in bioinformatics, Data management: Ability to use version control systems for data analysis projects", "Network Science: Understanding of graphs as representations of complex systems, \"Network Science: Understanding of nodes, edges, degree and paths in networks\", Network Science: Ability to construct network representations from tabular data, \"Network Science: Ability to calculate basic network measures such as degree distribution, centrality and clustering coefficient\", Network Science: Ability to visualise networks using appropriate layouts, Network Science: Ability to interpret network measures in the context of bioscience or medical case studies", "Machine Learning: Understanding machine learning techniques for bioscience image data , Machine Learning: Understanding of the difference between supervised and unsupervised learning, \"Machine Learning: Understanding of training, validation and test sets\", Machine Learning: Ability to train and evaluate classification models using scikit-learn, Machine Learning: Ability to train and evaluate regression models using scikit-learn, Machine Learning: Ability to apply clustering methods such as k-means, Machine Learning: Ability to use dimensionality reduction techniques such as PCA, Machine Learning: Ability to use cross-validation for model evaluation, \"Machine Learning: Ability to interpret common evaluation metrics such as accuracy, precision, recall and ROC AUC\", Machine Learning: Ability to identify and mitigate overfitting", null, null, null, null, null, null, "Classical statistics:", null, "Python Programming: Ability to handle basic errors and exceptions in Python code", "R programming:, Classical statistics: Understanding that variables have distributions, Classical statistics: Understanding independent/expanatory variables and dependent/response variables, \"Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous\"", null, null, null, "R programming: Proficiency in data manipulation and visualization", null, null, null, "R programming: Proficiency in data manipulation and visualization, \"Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous\"", "R programming: Proficiency in data manipulation and visualization", "\"Python Programming: Ability to use variables and basic data types such as integers, floats, strings and booleans\", Python Programming: Ability to work with Python libraries for scientific computing such as NumPy and pandas", "Bioinformatics: Understand the principles of NGS data pre-processing and genome alignment", "Bioinformatics: Understand the principles of NGS data pre-processing and genome alignment", "Bioinformatics: Understand the principles of NGS data pre-processing and genome alignment", "Bioscience: Understanding of DNA and genes"], "contributor": [null, null, null, null, null, null, null, "Meechan, Kimberly; Thompson, Stephen; Prins, Stella", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null], "copyrightHolder": ["Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Health+BioscienceIDEAS", "dataCAMPP, FutureLearn", "King's College London", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "ELIXIR-UK", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "Learn to Discover (L2D)", "dataCAMPP, FutureLearn", "dataCAMPP, FutureLearn", "dataCAMPP, FutureLearn", "dataCAMPP, FutureLearn", null, "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London", "King's College London"], "copyrightNotice": ["Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "Licensed under CC-BY 4.0 2021\u201324 by Cloud-SPAN", "CC-BY 4.0", null, null, "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "\u00a9 Learn to Discover (L2D). All rights reserved.", "\u00a9 Learn to Discover (L2D). All rights reserved.", "\u00a9 Learn to Discover (L2D). All rights reserved.", "\u00a9 Learn to Discover (L2D). All rights reserved.", "\u00a9 Learn to Discover (L2D). All rights reserved.", "FutureLearn. All rights reserved. ", "FutureLearn. All rights reserved. ", "FutureLearn. All rights reserved. ", "FutureLearn. All rights reserved. ", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null], "copyrightYear": [2022.0, 2023.0, 2021.0, 2022.0, 2023.0, 2021.0, 2021.0, 2025.0, 2022.0, 2022.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2022.0, 2021.0, 2022.0, 2022.0, 2020.0, 2022.0, 2022.0, 2025.0, 2025.0, 2025.0, 2025.0, 2025.0, null, null, null, null, null, 2024.0, 2022.0, 2024.0, 2023.0, 2023.0, 2022.0, null, 2023.0, 2023.0, 2022.0, 2024.0, 2024.0, 2022.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0, 2023.0], "creditText": ["Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E. (2022). Cloud-SPAN Metagenomics Course: Overview (Version 1.0.0) [Computer software]. https://doi.org/10.5281/zenodo.7505859", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E. (2024). Cloud-SPAN Metagenomics Course (2024 version, Environmental Applications) (Version 1.0.0) [Computer software]. https://doi.org/10.5281/zenodo.12700160", "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E. (2022). Cloud-SPAN Prenomics Course (Version 0.1.0) [Computer software]. https://doi.org/10.5281/zenodo.6354056", "Rand, E. & Greeves, E. (2023). Cloud-SPAN Core R Workshop [Training material]. Cloud-SPAN, University of York. Retrieved from https://cloud-span.github.io/core-r/02_intro_to_r_and_working_with_data.html", "Buenabad-Chavez, J., Rand, E., & Chong, J. (2024). Cloud-SPAN \u201cCreate Your Own AWS Instance\u201d Course [Training material]. Cloud-SPAN, University of York. Retrieved from https://cloud-span.github.io/create-aws-instance-0-overview", "Rand, E., Buenabad-Chavez, J., Chong, J., Forrester, S., & Greeves, E. (2022). Cloud-SPAN Genomics Course Overview (Version 1.0.0) [Computer software]. https://doi.org/10.5281/zenodo.6564314", "Rand, E., & colleagues. (2023). Cloud-SPAN Statistically Useful Experimental Design Course [Training material]. Cloud-SPAN, University of York. Retrieved from https://cloud-span.github.io/experimental_design00-overview", "Health+BioscienceIDEAS", "French, Andrew; Pridmore, Tony; Mellor, Nathan", "Roberts, Angus; Stahl, Daniel; Abdullah, Zahra; Langham, John;", "Sandra Ng, Tim Beck, Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Sonal Henson, Munazah Andrabi, Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Laura Cooper, Julia Jakiela, Katarzyna Kamieniecka, Krzysztof Poterlowicz", "Julia Jakiela, Wendi Bacon ", "Laura Coope, Julia Jakiela, Katarzyna Kamieniecka, Krzysztof Poterlowicz", "Edward Parkinson, Munazah Andrabi, Katarzyna Kamieniecka", "Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Katarzyna Kamieniecka, Krzysztof Poterlowicz ", "Richard Ostler", "Robert Andrews, Nick Juty, Munazah Andrabi, Nicola Soranzo, Sara Morsy, Kellie Snow, Korneel Hens, Philippe Rocca-Serra, Laura Cooper, Xenia Perez Sitja, Andrew Mason, Branka Franicevic, Saskia Lawson-Tovey, Krzysztof Poterlowicz, Katarzyna Kamieniecka", "Grimes, G. R., Floden, E., Di Tommaso, P., Ewels, P., & Garcia, M. (2021). Introduction to Workflows with Nextflow and nf-core. https://github.com/carpentries-incubator/workflows-nextflow", "Booth, T., Medd, N., Dunn-Davies, H., Emelianova, K., & Turner, F. (2021). Bioinformatics workflows with Snakemake. https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/", "Zieli\u0144ski, T. & Romanowski, A. (2022). FAIR in (biological) practice. https://carpentries-incubator.github.io/fair-bio-practice/", "O'Callaghan, A., Ewing, A., & Vallejos, C. (2022). High-dimensional statistics with R. https://carpentries-incubator.github.io/high-dimensional-stats-r/", "Pugh, D. R. (2020). Introduction to Conda for (Data) Scientists. https://carpentries-incubator.github.io/introduction-to-conda-for-data-scientists/", "Pollard, Tom; Peru, Giacomo (2022). Introduction to Machine Learning in Python. https://carpentries-incubator.github.io/machine-learning-novice-python/", "Herman, E., Rand, E., & Stewart, A. (2022). Statistical analysis for public health. https://carpentries-incubator.github.io/statistics-r-public-health/", null, null, null, null, null, "Atkinson, Jonathan; Postnikov, Andrey; Gaju, Oorbessy", "French, Andrew; Parsons, Simon", "Pound, Michael; French, Andrew; Mellor, Nathan", "Pound, Michael; Mellor, Nathan", "Pound, Michael; French, Andrew; Mellor, Nathan", "Cummins, Nicholas; Dinely, Judith; Stahl, Daniel; Abdullah, Zahra; Langham, John;", "Carr, Ewan; Ginestet, Cedric; Abdullah, Zahra; Langham, John; Stahl, Daniel;", "Flaviani, Flavia; Vigilante, Alessandra", "Ibrahim, Zina; Stahl, Daniel; Abdullah, Zahra; Langham, John; Qian, Linglong; Lehchevska, Nicole;", "Iniesta, Raquel; Agbedjro, Deborah; Shamsutdinova, Diana; Ginestet, Cedric; Abdullah, Zahra; Langham, John; Stahl, Daniel", "Cardoso, Jorge; Robinson, Emma; King, Andrew; Young, Alistair; Booth, Thomas; Da Silva, Mariana;", null, "Modat, Mark; Kerfoot, Eric; Barfoot, Theodore; Prokopenko, Denis; Cardoso, Jorge", "Hamrud, Eva; Thiery, Alex;", "Vigilante, Alessandra; Belokhvostova, Dasha; Cain, James;", "Roberts, Angus; Velupillai, Sumithra;", "Iniesta, Raquel; Dineley, Judith; Abdullah, Zahra; Vitoratou, Silia; Stahl, Daniel; Lehchevska, Nicole;", "Vigilante, Alessandra; Belokhvostova, Dasha; Cain, James;", "Hamid, Fursham; Lozano Gonzalez, Karla", "Cardoso, Jorge; Robinson, Emma; Da Silva, Mariana; Borges, Pedro; Langham, John; Crawford, Andre;", "Vigilante, Alessandra;", "Vigilante, Alessandra; Hamrud, Eva; Thiery, Alex;", "Hamrud, Eva; Thiery, Alex;", "Vigilante, Alessandra;"], "description": ["Learn to assemble, quality-check, and bin metagenomes into MAGs, assign taxonomy, calculate diversity, and perform functional annotation using mock microbial community data. No prior bioinformatics experience required; basic microbiome knowledge recommended.", "Learn to assemble, quality-check, and bin metagenomes into MAGs, assign taxonomy, calculate diversity, and perform functional annotation using real soil microbial community data. No prior bioinformatics experience required; basic microbiome knowledge recommended.", "Develop foundational skills in AWS, Linux command line (Bash), and basic cloud-based computing for omics data analysis.", "Gain a basic proficiency in R for data manipulation, visualization, and statistical analysis. Covers RStudio navigation, data types, data import, summarisation, and plotting.", "Set up and manage a personal AWS virtual machine instance for cloud-based bioinformatics analysis. Covers AWS account setup, instance lifecycle, and cost management.", "An introduction to genomics, covering topics such as sequencing, variant calling, and data processing.", "Understand the principles of good experimental design for biological research, enabling robust and reproducible statistical analysis.", "Introduction to the fundamentals of imaging data that has been acquired through light microscopy experiments and usage of Napari.", "This six-week course from the University of Nottingham will introduce you to the fundamentals of image analysis and its applications in plant phenotyping. You\u2019ll learn how to use imaging technologies to collect data from images and perform detailed analyses. By the end of the course, you\u2019ll be able to extract meaningful information from image data without any destruction or harm to the subjects of your study.", "By the end of this module, you will be able to use Python to import and manipulate datasets, and be confident in performing programming operations such as using conditionals, loops, and functions, as well as conducting basic statistical analyses.", "In this tutorial, you will learn the benefits of making clinical datasets FAIR (Findable, Accessible, Interoperable, and Reusable) and how to make them FAIR.", "In this Galaxy Training Network tutorial you will lear To populate ENA metadata objects through the Webin portal and to submit raw reads to ENA using FTP", "This tutorial is a short introduction to the FAIR data management framework.\r\n\r\n", "In this Galaxy Training Network tutorial you will organise bioimage metadata, find out what REMBI is and why it is useful, categorise what metadata belongs to each of the submodules of REMBI and gather the metadata for an example bioimage dataset", "This is a Galaxy tutorial to use Scanpy for single cell analysis\r\n\r\n", "In this Galaxy Training Network tutorial you will locate bioimage data repositories, compare repositories to find which are suitable for your data and find out what the requirements are for submitting", "This lesson provides a practical guide to sourcing and pre-processing a bulk RNA-Seq dataset for use in a machine learning classification task. The lessons explains the characteristics of a dataset required for this type of analysis, how to search for and download a dataset from each of the main public functional genomics repositories, and then provides guidelines on how to pre-process a dataset to make it machine learning ready, with detailed examples. The lesson finally explains some of the additional data filtering and transformation steps that will improve the performance of machine learning algorithms using RNA-Seq count data.\n\nThe lesson is written in the context of a supervised machine learning classification modelling task, where the goal is to construct a model that is able to differentiate two different disease states (e.g. disease vs. healthy control) based on the gene expression profile.", "This tutorial outlines how to set and use existing resources to make Galaxy training development and content creation FAIR.\n", "This tutorial is a short introduction to the FAIR principles and their origin. You can find out more at the FAIR Pointers online course.\r\n\r", "This lesson is aimed at long-term agricultural experiment (LTE) managers, and researchers interested in using long-term agricultural experiment data, who want to apply FAIR principles and good research data management practices to their LTE data. While the target audience for this lesson is therefore LTE managers and researchers, the lesson covers many data management and FAIR data topics relevant to a broader audience working on agricultural field trials.", "This short Carpentries-style course caters for people working in the Life Sciences with little or no experience with the FAIR principles (Findable, Accessible, Interoperable, Reusable). The course aims to concisely introduce FAIR, its concepts and terminology and supplement all material with signposting to helpful FAIR resources. These resources include RDMkit, FAIR Cookbook, RDMbites, FAIRsharing and the Data Stewardship Wizard, offered through the ELIXIR open-access infrastructure, supporting FAIR research data management.", "Introduction to bioinformatics workflows with Nextflow and nf-core. Nextflow is a workflow management system for building and sharing computational pipelines that facilitate reproducible data science workflows. nf-core is a community-led repository of high quality bioinformatics workflows using Nextflow.", "Learn to create reproducible and scalable bioinformatics workflows with Snakemake. Covers core Snakemake syntax, chaining rules, parallel execution, and Conda integration using RNA-seq analysis examples.", "Hands-on workshop on Open Science and FAIR (Findable, Accessible, Interoperable, Reusable) principles for biological research. Covers practical application of FAIR principles throughout project life cycles, including data management, metadata, file organisation, and data sharing.", null, "Introduction to Conda package and environment management system for reproducible data science workflows. Covers creating, saving, loading, and switching between environments on Windows, macOS and Linux for Python, R, and other languages.", "Applied machine learning in Python using health data. Covers data preparation, resampling, model training, and performance evaluation including decision trees, neural networks, and responsible machine learning practices.", "Statistical analysis in R covering statistical thinking, simple linear regression, multiple linear regression and logistic regression. Examples use public health data but methods are relevant to life, health and social sciences researchers.", "A comprehensive introduction to programming in Python for learners with limited or no prior experience. The module covers core programming concepts, algorithmic thinking, input/output, variables and data types, logical operations, conditionals, iterations, functions, basic data structures, and working with arrays, lists, tuples and strings. It prepares learners for data-focused work and is a prerequisite for subsequent L2D modules in Data Handling and Machine Learning.", "A hands-on introduction to data handling in Python, building directly on the Python Fundamentals module. Learners work with NumPy arrays and pandas DataFrames to import, clean, manipulate, analyse and visualise numerical, tabular and image data. Topics include univariate and multivariate analysis, basic statistics, plotting with Matplotlib, and good practice in data workflows. Successful completion is a compulsory prerequisite for the Machine Learning module.", "A practical introduction to network science for data-rich bioscience and medical applications. Learners explore graphs and network representations of complex systems, network measures and quantification, and network visualisation. The module focuses on applying network analysis techniques to real-world case studies, with an emphasis on interpreting results in a research context.", "An applied machine learning module building on Python programming and data handling skills. Learners implement supervised and unsupervised machine learning methods using scikit-learn, including classification algorithms, clustering and dimensionality reduction. The module covers model evaluation, performance improvement and robustness, and emphasises understanding why models work and what they reveal about data.", "An advanced module that builds on the Machine Learning component of the L2D curriculum. The module introduces core concepts in artificial intelligence, focusing on three primary AI paradigms (for example, logical/symbolic approaches, probabilistic methods and modern neural approaches). Learners examine how AI systems are constructed, their applications in research, and the broader implications of AI in bioscience and medicine.", "On this four-week course, you\u2019ll delve into some of the various technologies used in plant phenotyping, learning about data generation in different environments \u2013 both the advantages and challenges. By the end of the course, you\u2018ll be able to demonstrate understanding of plant phenotyping concepts on several levels: from basic principles to modern state, challenges, and different robot kits for data collection.", "On this five-week course from the University of Nottingham, you\u2019ll gain an overview of the applications of machine learning for image data, focusing specifically on its use in plant phenotyping.", "This course is designed to bridge the gap between theoretical knowledge and practical application, providing a comprehensive understanding of how experimental design principles can optimise machine learning outcomes. By the end of this course, you\u2019ll have a deep understanding of how experimental design supports machine learning, driving innovation in biosciences.", "On this five-week course, you\u2019ll develop essential knowledge of deep learning to understand how it can help your research in bioscience. You\u2019ll be guided through deep learning techniques and gain practical skills you can use outside of the course. Throughout the course, you\u2019ll take part in practical exercises to cement your knowledge and ensure you can use your skills in your context.", "On this five-week online course with the University of Nottingham, you\u2019ll dive into the essentials of generative AI, including large language models (LLMs) and image generation methods. You\u2019ll learn how to harness AI to address challenges like missing data and generate insightful visualisations to enhance your research. Through practical examples, you\u2019ll explore how AI can be integrated into bioscience workflows, empowering you to make significant advancements in your field.", "\nThis course introduces you to the fundamental techniques and concepts necessary for understanding and implementing AI applications in healthcare. AI concepts covered include; an introduction to intelligent agents; problem-solving strategies such as evolutionary algorithms and particle swarm optimisation; constraint satisfaction problems; modelling uncertainty; and, learning methods including deep learning and reinforcement learning.\n\nThroughout this course, you will gain valuable hands-on experience by implementing AI techniques, using real healthcare data examples, in Python as well as learning to assess and validate the resulting models.", "This module introduces you to the basics (and beyond) of the R statistical programming language, and gets you started working with it, in a health research context. You will benefit from having a background in basic statistical methods (e.g. probability distributions, hypothesis testing) and high school mathematics (e.g. basic calculus, basic linear algebra)", null, null, "Prediction modelling is distinct from statistical modelling in that the primary objective is to deliver models that make accurate predictions, rather than estimate parameters such as risk factors. This course is for you if you want to:\nHave a good understanding of core clinical prediction concepts, such as prognosis, prognostic factors, prognostic models, and stratified medicine and want to be able to apply this understanding to the design, conduct, and interpretation of clinical prediction modelling research studies, and the critical appraisal of research papers.\nBe able to describe how modern statistical concepts, regression and machine learning methods can be applied to medical prediction problems.\nBe familiar with the principles that play a role in internal validation such as over-fitting, optimism and shrinkage and understand key components of internal validation methods such as cross-validation and bootstrapping.\nBe able to develop simple prediction models, assess their quality and validate them using R software\nBe able to critically assess the general applicability of a developed model to predict future outcomes.\nBe equipped with a range of statistical and machine learning skills, which will enable you to take prominent roles in a wide spectrum of employment and research.", "Like in other industries, there has been a growing investment in the use of Artificial Intelligence (AI) and Machine Learning in health care to help solve clinical, research and service provision challenges. With so much information currently available it is hard to know exactly what is meant by all the various terms in the field. This introductory course will help you understand exactly what AI and machine learning are, and importantly what they are not. You will be able to see real life examples where AI has been applied in the healthcare setting and have time to consider where you might interact with it. This module is the perfect starting point to dive into the world of AI within Healthcare.  ", "A good experimental design is the first step for a successful bioinformatics analyses, the second one is to understand how to record and organise data. Most life scientists (at any career stage) store data in spreadsheets, so this is the place that many research projects start and how data are delivered to the bioinformatician. Participants will prepare data for bioinformatics analyses and improve their data organisation skills.", "Leading on from Demystifying AI or for participants with a base level understanding of AI already this course will give participants an introduction to writing their own code to be used on AI platforms. Over four weeks participants will be taught Python fundamentals including:\n\nData visualisation\nLoops\nFunctions\nErrors and exceptions\nDebugging\nData I/O", "Welcome to our Next-Generation Sequencing (NGS) Introduction course where you will gain a comprehensive understanding of how NGS work, how different types of NGS can answer different biological questions and how the raw data processing is in common between the different platforms with a particular focus to bulk RNA-seq and ATAC-seq. This introduction is preparatory for our RNA-seq, ATAC-seq and variant analysis using Galaxy courses where we will go through the specific analyses after the pre-processing steps. ", "R is one of the most used programming languages in Data Science to perform data analysis, used to visualize the data, to apply statistics and for machine learning analyses. This course is designed for participants with no programming experience. We will be using example questions related to health research to learn how to write ad-hoc codes to manipulate and visualize data using R packages designed for Big Data analyses.", "Much of the information in Electronic Health Records (EHRs) is held in unstructured, natural language text. This is usually in the form of correspondence between clinicians, or short notes added in to the EHR. If we wish to reuse EHRs for research, then we need ways to handle this text. \n\nThis course explores Natural Language Processing (NLP), and its use in extracting this information from EHR text. The course aims to equip you with an understanding of NLP and its techniques, in particular when applied to biomedical texts such as the unstructured, free text portion of medical records, in order that you can: \n\nDecide when it is appropriate to use NLP to help with the analysis of medical records and other biomedical text, in terms of: \n- Costs and Benefits \n- Appreciating the data and language specific to the health domain, including what data is available, and what is needed in order to extract the information required. \nImplement NLP solutions based on this decision.\nEvaluate the solution. \nInvestigate ways in which NLP science can be extended to improve performance and tackle new challenges.", "This course provides students with a sound introduction to statistical principals and methods with an emphasis on conceptual understanding rather than learning formulas and a cookbook approach of statistical tests. The course includes introductory statistical techniques including core descriptive and inferential concepts for a strong foundation grounding for learning statistical modelling approaches to research.  \n\nThe course will present diverse methods of analysis including comparing groups, regression, mediation and moderation, as variants of the single underlying strategy of fitting a linear model to data and the generalised linear model for categorical data. Emphasis will be on selecting an appropriate form of analysis for a given research design and the interpretation and communication of results.  \n\nLearners are also introduced to computer-based statistical software exploration and analysis using real-life examples from psychology and psychiatry using the software SPSS. ", null, "Single-cell transcriptomics is a powerful tool to study the heterogeneity of the cellular transcriptome at single-cell levels. This course aims to equip participants with essential skills to process RNA-sequencing reads from single-cell transcriptomics experiment and to perform downstream analyses of the single-cell gene expression data. This course is designed for participants with some experience in R programming and running command-line programs. We will be using published single-cell RNA-sequencing datasets to perform quality control, data normalisation, cell clustering, differential expression and trajectory analyses.", "The goal of this course is to equip you with sufficient foundational knowledge, but primarily practical coding and engineering skills to ensure that you feel confident applying deep learning to real world health care problems and adapting to the fast paced field as it moves forward in future.\n\nThis is an intermediate level course, requiring you to have a working knowledge of Python, including classes and the ability to work with packages such as numpy. It will help you to have studied mathematics at high school, since this course uses concepts such as first order partial derivatives and matrix algebra in its teaching.\n\nThis course teaches how to work with image data, which has a richer structure than many other types of data. Thus it is important for you to take the time to understand theses structures, and how they are transformed prior to being used in deep learning models.", "The aim of this course is designed for students without programming experience that want to process and analyse NGS data. In this course, we will use Galaxy, an open-source, web-based platform for accessible, reproducible, and transparent computational biomedical research. Even if we are only going to cover an NGS pipeline here (variant calling), this course will give students enough understanding on how Galaxy works, allowing them to easily run bioinformatics tools, setting up parameters and building larger workflows.\n\nBefore taking this course, it is important to understand the basics of NGS by taking the NGS: Introduction and pre-processing pipeline course", "Welcome to our NGS: RNA-seq course where you will gain a comprehensive understanding of how to analyse and visualise bulk RNA-seq data. Before taking this course, it is important to understand the basics of NGS by taking the NGS: Introduction and pre-processing pipeline course.", "Welcome to our NGS: ATAC-seq course where you will gain a comprehensive understanding of how to analyse and visualise ATAC-seq data. \n\nBefore taking this course, it is important to understand the basics of NGS by taking the NGS: Introduction and pre-processing pipeline course.", "The aim of this course is to provide participants with a comprehensive understanding of sequence alignment techniques, focusing on both BLAST and multiple sequence alignment, to equip them with the skills necessary to analyze biological sequences, uncover evolutionary relationships, identify functional elements, and explore the vast applications in fields such as genomics, proteomics, and bioinformatics."], "educationalAlignment": [null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, "Statistical and computational methods for extremely high-dimensional data in biomedical and health sciences (gene expression, DNA methylation, health records). Covers high-dimensional regression, dimensionality reduction, and clustering.", null, null, null, "Introductory Python programming for data science in biosciences and medicine", "Introductory data handling and visualisation for data science in biosciences and medicine", "Introductory network science for data science in biosciences and medicine", "Applied machine learning for data science in biosciences and medicine", "Advanced artificial intelligence concepts and applications for research", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null], "educationalLevel": ["level1", "level1", "level1", "level1", "level1", "level2", "level1", "level1", "level3", "level2", "level1", "level1", "level1", "level1", "level2", "level2", "level3", "level2", "level2", "level1", "level1", "level2", "level2", "level1", "level2", "level1", "level2", "level1", "level1", "level2", "level2", "level2", "level3", "level2", "level4", "level3", "level5", "level3", "level1", "level1", "level1", "level2", "level2", "level1", "level1", "level3", "level2", "level1", "level1", "level1", "level2", "level2", "level3", "level2", "level2", "level2", "level2"], "featured": ["no", "no", "no", "yes", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no"], "featuredText": ["no", "no", "no", "yes", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no", "no"], "funder": ["UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI", "UKRI"], "projectFunding": ["Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1)", "Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1), Natural Environment Research Council (NE/X006999/1, E/Y003527/1).", "Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1)", "Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1)", "Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1)", "Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1)", "Cloud-SPAN. Funded by the UKRI Innovation Scholars (MR/V038680/1)", "Health and bioscience IDEAS. Funded by UKRI Innovation Scholars (MR/V03863X/1). ", "Data CAMPP. Funded by UKRI Innovation Scholars (MR/V038850/1).", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "ELIXIR-UK: FAIR data stewardship training. Funded by UKRI Innovation Scholars (MR/V038966/1).", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Ed-DASH Funded by UKRI Innovation Scholars (MR/V039075/1). ", "Learn to Discover (L2D). Funded by UKRI Innovation Scholars", "Learn to Discover (L2D). Funded by UKRI Innovation Scholars", "Learn to Discover (L2D). Funded by UKRI Innovation Scholars", "Learn to Discover (L2D). Funded by UKRI Innovation Scholars", "Learn to Discover (L2D). Funded by UKRI Innovation Scholars", "Data CAMPP. Funded by UKRI Innovation Scholars (MR/V038850/1).", "Data CAMPP. Funded by UKRI Innovation Scholars (MR/V038850/1).", "Data CAMPP. Funded by UKRI Innovation Scholars (MR/V038850/1).", "Data CAMPP. Funded by UKRI Innovation Scholars (MR/V038850/1).", "Data CAMPP. Funded by UKRI Innovation Scholars (MR/V038850/1).", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). ", "Innovation Scholars Big Data Skills Training. Funded by UKRI Innovation Scholars (MR/V038664/1). "], "identifier": ["metagenomics", "metagenomics_env", "prenomics", "core_r", "aws_instance", "genomics", "experimental_design", "microscopy-novice", "introduction-to-image-analysis-for-plant-phenotyping", "introduction-to-python-for-health-research", "gtn-tutorial-fair-clinical-data", "gtn-tutorial-ena-sequence-submission", "gtn-tutorial-fair-data-management-solutions", "gtn-tutorial-rembi-metadata", "gtn-tutorial-single-cell", "gtn-tutorial-bioimage-metadata-course", "carpentries-course-rna-seq-ml", "gtn-tutorial-fair-galaxy-training", "gtn-tutorial-fair-in-a-nutshell", "carpentries-course-agricultural-data-long-term", "carpentries-course-fair-pointers", "nextflow", "snakemake", "fair-bio-practice", "high-dim-stats-r", "intro-conda", "ml-python", "intro-stats-r", "L2D-PYTHON-FUNDAMENTALS-2025", "L2D-DATA-HANDLING-2025", "L2D-NETWORK-SCIENCE-2025", "L2D-MACHINE-LEARNING-2025", "L2D-ARTIFICIAL-INTELLIGENCE-2025", "plant-phenotyping-technologies", "machine-learning-for-image-data", "experimental-design-for-machine-learning", "deep-learning-for-bioscientists", "generative-ai-for-plant-phenotyping", "ai-in-health-analytics", "introduction-to-r-for-health-research", "unix", "machine-learning-for-health-research", "prediction-modelling-for-health-research", "demystifying-ai", "using-spreadsheets", "intermediate-python-and-software-engineering", "ngs-introduction-and-pre-processing-pipeline", "basic-r-with-data-carpentry", "natural-language-processing-for-health-research", "introduction-to-statistics", "statistics-with-r", "ngs-single-cell-rna-sequencing-analysis", "applied-artificial-intelligence-for-health-research", "ngs-on-the-galaxy-platform", "next-generation-sequencing-rna-seq", "ngs-atac-seq", "sequence-alignment-blast"], "inLanguage": ["EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN", "EN"], "isAccessibleForFree": ["yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "no", "no", "no", "no", "no", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes", "yes"], "isBasedOn": [null, "Buenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E. (2022). Cloud-SPAN Prenomics Course (Version 0.1.0) [Computer software]. https://doi.org/10.5281/zenodo.6354056\nBuenabad-Chavez, J., Cansdale, A., Chong, J., Forrester, S., Greeves, E., & Rand, E. (2022). Cloud-SPAN Metagenomics Course: Overview (Version 1.0.0) [Computer software]. https://doi.org/10.5281/zenodo.7505859", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, "https://seqera.io/training/, https://nf-co.re/usage/tutorials/nf_core_tutorial", null, null, null, null, null, null, null, "https://learntodiscover.ai/product/basic-python-2025/", "https://learntodiscover.ai/product/basic-python-2025/", "https://learntodiscover.ai/product/basic-python-2025/; https://learntodiscover.ai/product/__trashed/", "https://learntodiscover.ai/product/machine-learning/", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, "https://lessons.datacarpentry.org/R-ecology-lesson/", null, null, null, null, null, null], "keywords": ["data import, programming, AWS, EC2, virtual machine, cloud computing, bioinformatics, data processing, metagenomics, HPC, QC, binning, taxonomy, annotation, bash, Linux, biological research, reproducibility, Metagenome assembly, Taxonomic assignment, Flye, Long reads, FASTQ, Pavion, Krona", "metagenomics, environmental science, HPC, QC, binning, taxonomy, annotation, bash, Linux, cloud computing, bioinformatics, data visualisation, RStudio, R, data import, programming, AWS, EC2, data processing, biological research, reproducibility, experimental design, data summarisation, statistics, Metagenome assembly", "AWS, bash, Linux, cloud computing, bioinformatics", "R, RStudio, data import, data visualisation, data summarisation, statistics, programming", "AWS, EC2, virtual machine, cloud computing, bioinformatics", "genomics, variant calling, data processing, bioinformatics", "experimental design, statistics, replication, reproducibility, biological research", "bioinformatics, microscopy, python, napari", "image analysis, plant phenotyping", "programming, python", "FAIR, Clinical Dataset, Dataset, Metadata", "genomics, bioinformatics, data processing, FASTA, FASTQ, Short reads, Long reads, Dataset, Metadata, FAIR, reproducibility", "FAIR, Dataset, Metadata, Clinical Dataset, Open Science, data stewardship, data management planning, data sharing, persistent identifiers, data licensing, reproducibility, biological research", "image analysis, microscopy, FAIR, Metadata, Dataset, data stewardship, data management planning, reproducibility", "genomics, bioinformatics, data processing, statistics, data visualisation, Dataset, Metadata, FAIR, reproducibility", "image analysis, microscopy, FAIR, Metadata, Dataset, data stewardship, data management planning, reproducibility", "genomics, bioinformatics, data processing, FASTQ, Short reads, statistics, machine learning, scikit-learn, supervised learning, classification, regression, model evaluation, cross-validation, feature engineering, Dataset, Metadata, reproducibility", "FAIR, Dataset, Metadata, Open Science, data stewardship, data management planning, data sharing, persistent identifiers, data licensing, reproducible workflows, workflow management, reproducibility", "FAIR, Dataset, Metadata, Open Science, data stewardship, data management planning, data sharing, persistent identifiers, data licensing", "Dataset, Metadata, FAIR, data stewardship, data management planning, data sharing, persistent identifiers, data licensing, reproducibility, biological research, plant phenotyping", "FAIR, Dataset, Metadata, Clinical Dataset, Open Science, data stewardship, data management planning, data sharing, persistent identifiers, data licensing, reproducibility", "bioinformatics, data processing, HPC, cloud computing, bash, Linux, python, Nextflow, nf-core, workflow management, reproducible workflows, Docker, Singularity, containers, pipeline development", "bioinformatics, data processing, HPC, bash, Linux, python, workflow management, workflow rules, Snakefile, job scheduling, workflow reproducibility, DAG (directed acyclic graph), reproducibility", "FAIR, Clinical Dataset, Metadata, Dataset, reproducibility, biological research, Open Science, data stewardship, data management planning, data sharing, persistent identifiers, data licensing", "R, RStudio, statistics, data summarisation, high-dimensional data, multivariate statistics, principal component analysis, regularised regression, factor analysis, clustering, k-means, hierarchical clustering", "python, data processing, Conda, environment management, package management, virtual environments, dependency management, reproducible environments", "python, machine learning, statistics, data visualisation, data summarisation, scikit-learn, supervised learning, unsupervised learning, classification, regression, model evaluation, cross-validation, feature engineering", "R, RStudio, data import, data visualisation, data summarisation, statistics, public health, epidemiology, health data, statistical modelling, regression analysis, hypothesis testing", "python, programming, data import, data visualisation, data summarisation, statistics, bash, Linux, Unix", "python, R, RStudio, data import, data visualisation, data summarisation, statistics, data processing, Dataset, Metadata, FAIR, reproducibility, biological research", "python, statistics, data visualisation, data summarisation, machine learning", "python, data visualisation, data summarisation, statistics, machine learning, deep learning, scikit-learn, supervised learning, unsupervised learning, classification, regression, model evaluation, cross-validation, feature engineering, pytorch", "python, statistics, machine learning, deep learning, generative AI, pytorch, data processing", "plant phenotyping", "machine learning, image analysis, plant phenotyping", "machine learning, plant phenotyping, experimental design", "deep learning, pytorch", "generative AI, machine learning, plant phenotyping", "programming, data processing, deep learning, python", "R, RStudio, data visualisation, statistics, programming", "Unix", "python, programming, Clinical Dataset", "statistics, R, RStudio, Clinical Dataset", "generative AI, deep learning, machine learning, AI", "data processing, Metadata, Clinical Dataset, Dataset, health data", "python, bash, Docker", "bioinformatics", "R, RStudio", "natural language processing", "statistics, statistical modelling, regression analysis", "R, RStudio, statistics, statistical modelling", "RStudio, NGS", "python, deep learning, machine learning, AI", "NGS, bioinformatics", "NGS, bioinformatics, RStudio", "NGS, bioinformatics, RStudio", "bioinformatics, RStudio"], "learningResourceType": ["onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "course, onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse", "onlineCourse"], "license": ["CC-BY 4.0", "CC-BY 4.0", "CC-BY 4.0", "CC-BY 4.0", "CC-BY 4.0", "CC-BY 4.0", "CC-BY 4.0", "CC-BY 4.0", null, null, "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", "CC BY 4.0", null, null, null, null, null, null, null, null, null, null, null, null, "CC-BY 4.0", null, null, null, null, null, null], "maintainer": ["Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "IDEAS", "data CAMPP", "Big Data Revolution", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "L2D", "L2D", "L2D", "L2D", "L2D", "data CAMPP", "data CAMPP", "data CAMPP", "data CAMPP", "data CAMPP", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution"], "provider": ["Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "Cloud-SPAN", "IDEAS", "data CAMPP", "Big Data Revolution", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "ELIXIR", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "Ed-DaSH", "L2D", "L2D", "L2D", "L2D", "L2D", "data CAMPP", "data CAMPP", "data CAMPP", "data CAMPP", "data CAMPP", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution", "Big Data Revolution"], "schemaVersion": ["Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15", "Schema.org \u2022 V29.2 | 2025-05-15"], "sdPublisher": ["Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org", "Schema.org"], "teaches": ["Computing: Understanding of hierarchical filesystems including root and home directories, Command line: Ability to specify inputs and outputs for software from command line, Command line: Ability to move files between local and remote systems from the command line, Command line: Ability to create and run shell scripts, Bioinformatics: Ability to interpret FASTQ Phred quality scores and FastQC/MultiQC plots, Bioinformatics: Ability to assess read quality using FastQC, Bioinformatics: Ability to trim and quality filter reads using cutadapt, Bioinformatics: Abllity to assemble a metagenome using Flye and Pilon, Bioinformatics: Ability to interpret assembly quality metrics (N50, contig lengths), Bioinformatics: Ability to bin MAGs using MetaBAT2, Bioinformatics: Ability to assign functional annotation to MAGs using Prokka, Bioinformatics: Ability to assign taxonomy to a MAG using kraken, Bioinformatics: Understanding of taxonomic assignment, Bioinformatics: Understanding of limitations of taxonomic assignment including database bias and completeness, Bioinformatics: Understanding of k-mers, Bioinformatics: Ability to explore functional annotations using BlastKOALA, Bioinformatics: Awareness of the concept of bioinformatics pipelines, Bioinformatics: Understanding of why a metagenome assembly polishing step is required, Bioscience: Understand sequencing sample quality in terms of sequencing depth and phred quality, High Performance Computing: Ability to manage cloud-based computing environments", "Computing: Understanding of hierarchical filesystems including root and home directories, Command line: Ability to specify inputs and outputs for software from command line, Command line: Ability to move files between local and remote systems from the command line, Command line: Ability to create and run shell scripts, Bioinformatics: Ability to interpret FASTQ Phred quality scores and FastQC/MultiQC plots, Bioinformatics: Ability to assess read quality using FastQC, Bioinformatics: Ability to trim and quality filter reads using cutadapt, Bioinformatics: Abllity to assemble a metagenome using Flye and Pilon, Bioinformatics: Ability to interpret assembly quality metrics (N50, contig lengths), Bioinformatics: Ability to bin MAGs using MetaBAT2, Bioinformatics: Ability to assign functional annotation to MAGs using Prokka, Bioinformatics: Ability to assign taxonomy to a MAG using kraken, Bioinformatics: Understanding of taxonomic assignment, Bioinformatics: Understanding of limitations of taxonomic assignment including database bias and completeness, Bioinformatics: Understanding of k-mers, Bioinformatics: Ability to explore functional annotations using BlastKOALA, Bioinformatics: Awareness of the concept of bioinformatics pipelines, Bioinformatics: Understanding of why a metagenome assembly polishing step is required, Bioscience: Understand sequencing sample quality in terms of sequencing depth and phred quality, High Performance Computing: Ability to manage cloud-based computing environments", "Command line: Ability to open a shell such as bash, Command line: Ability to navigate filesystems from the command line, High Performance Computing: Ability to manage cloud-based computing environments, High Performance Computing: Ability to utilize cloud computing resources for data analysis, Computing: Understanding of hierarchical filesystems including root and home directories, \"Computing: Understanding working directories, relative and absolute paths\", Computing: Understanding of File types and formats including plain text files and the relation between file types and file extensions, \"Command line: Ability to move, delete, create and copy files and directories from command line\", Command line: Ability to use common command line functions, Command line: Ability to specify inputs and outputs for software from command line, Command line: Ability to login to remote systems from the command line, Command line: Ability to move files between local and remote systems from the command line, Command line: Ability to create and run shell scripts, Data management: Ability to document and share data analysis workflows", "Computing: Understanding of File types and formats including plain text files and the relation between file types and file extensions, R programming: Ability to create and use dataframes, R programming: Ability to plot graphs and figures using ggplot2, R programming: Proficiency in data manipulation and visualization", "High Performance Computing: Ability to manage cloud-based computing environments, High Performance Computing: Ability to utilize cloud computing resources for data analysis, Command line: Ability to move files between local and remote systems from the command line, Command line: Ability to specify inputs and outputs for software from command line", "Bioinformatics: Understanding of genomic data analysis, Bioinformatics: Understanding of variant calling, Data management: Understanding of data integrity and quality control in bioinformatics", "Experimental Design: Understanding of experimental design principles in metagenomics, Experimental Design: Understanding of principles for robust and reproducible research, Experimental Design: Understanding foundations of experimental design for machine learning, Classical statistics: Understanding of high-dimensional data challenges, Classical statistics: Understanding of correlation and multicollinearity", "Image processing: Understanding of basics of image analysis, Bioscience: Plant phenotyping", "Image processing: Understanding of basics of image analysis", "Python Programming:", "Data management: Ability to get clinical annotation from data providers, Data management: Ability to make clinical datasets FAIR, Data management: Ability to ccheck the FAIRnes of the clinical records", "Bioinformatics: Understanding of genomic data analysis, Bioinformatics: Awareness of the concept of bioinformatics pipelines, FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing, Data management: Ability to document and share data analysis workflows, Data management: Understanding of data integrity and quality control in bioinformatics", "FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to organize data with appropriate naming conventions, FAIR and Open Science: Ability to use electronic lab notebooks, FAIR and Open Science: Ability to use data repositories, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing, Data management: Ability to document and share data analysis workflows, Data management: Ability to make clinical datasets FAIR, Data management: Ability to get clinical annotation from data providers, Data management: Ability to ccheck the FAIRnes of the clinical records", "FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to organize data with appropriate naming conventions, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing, Data management: Ability to document and share data analysis workflows, Data management: Understanding of data integrity and quality control in bioinformatics", "Bioinformatics: Comprehensive understanding of how to analyse and visualise bulk RNA-seq data, Bioinformatics: Comprehensive understanding of how to analyse and visualise ATAC-seq data, Bioinformatics: Process and analyse NGS data using Galaxy, Classical statistics: Ability to interpret PCA results, Classical statistics: Ability to evaluate clustering using silhouette scores and Dunn index, Machine Learning: Ability to apply clustering methods such as k-means, Machine Learning: Ability to use dimensionality reduction techniques such as PCA", "FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to organize data with appropriate naming conventions, FAIR and Open Science: Ability to use data repositories, FAIR and Open Science: Understanding of persistent identifiers, Image processing: Understanding of basics of image analysis, napari: Ability to use napary for image analysis", "Bioinformatics: Comprehensive understanding of how to analyse and visualise bulk RNA-seq data, Bioinformatics: Understanding of genomic data analysis, Classical statistics: Understanding of high-dimensional data challenges, Classical statistics: Ability to interpret PCA results, Machine Learning: Understanding ML methodologies, Machine Learning: Understanding of the difference between supervised and unsupervised learning, Machine Learning: Understanding of training, validation and test sets, Machine Learning: Ability to train and evaluate classification models using scikit-learn, Machine Learning: Ability to use dimensionality reduction techniques such as PCA, Machine Learning: Ability to use cross-validation for model evaluation, Machine Learning: Ability to interpret common evaluation metrics such as accuracy, precision, recall and ROC AUC, Machine Learning: Ability to identify and mitigate overfitting, Machine Learning: Understanding of dataset quality issues", "FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to organize data with appropriate naming conventions, FAIR and Open Science: Ability to use data repositories, FAIR and Open Science: Understanding of persistent identifiers, Data management: Ability to document and share data analysis workflows, Workflow Management: Understanding of workflow management systems, Workflow Management: Understanding of workflow reproducibility", "FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing, Data management: Understanding of data integrity and quality control in bioinformatics", "FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to organize data with appropriate naming conventions, FAIR and Open Science: Ability to use electronic lab notebooks, FAIR and Open Science: Ability to use data repositories, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing, Data management: Ability to document and share data analysis workflows", "FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to use data repositories, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing", "Workflow Management: Understanding of workflow management systems, Workflow Management: Understanding of Nextflow processes channels and workflows, Workflow Management: Ability to write Nextflow scripts, Workflow Management: Ability to use workflow parameters, Workflow Management: Understanding of parallel execution in workflows, Workflow Management: Ability to configure computational resources in workflows, Workflow Management: Ability to use nf-core community pipelines, Workflow Management: Understanding of workflow reproducibility, Data management: Ability to document and share data analysis workflows", "Workflow Management: Understanding of workflow management systems, Workflow Management: Ability to write Snakemake rules, Workflow Management: Understanding of wildcards and pattern matching in workflows, Workflow Management: Ability to chain rules in a workflow, Workflow Management: Understanding of DAG execution in workflows, Workflow Management: Ability to optimize workflow performance, Workflow Management: Understanding of workflow reproducibility, Workflow Management: Ability to handle complex inputs and outputs in workflows, Environment Management: Understanding of Conda integration with Snakemake, Data management: Ability to document and share data analysis workflows", "FAIR and Open Science: Understanding of Open Science principles, FAIR and Open Science: Understanding of FAIR principles, FAIR and Open Science: Understanding of metadata and ontologies, FAIR and Open Science: Ability to organize data with appropriate naming conventions, FAIR and Open Science: Ability to use electronic lab notebooks, FAIR and Open Science: Ability to use data repositories, FAIR and Open Science: Understanding of persistent identifiers, FAIR and Open Science: Understanding of data licensing, Data management: Ability to make clinical datasets FAIR, Data management: Ability to use version control systems for data analysis projects, Computing: Use Git repositories for collaboration and version control, Experimental Design: Understanding of principles for robust and reproducible research, Artificial Intelligence:", "Classical statistics: Understanding of high-dimensional data challenges, Classical statistics: Understanding of correlation and multicollinearity, Classical statistics: Ability to interpret PCA results", "Environment Management: Ability to create and manage Conda environments, Environment Management: Ability to install packages using Conda, Environment Management: Ability to export and share Conda environments, Environment Management: Understanding of environment reproducibility", "Machine Learning: Understanding of the difference between supervised and unsupervised learning, \"Machine Learning: Understanding of training, validation and test sets\", Machine Learning: Ability to prepare data for machine learning, Machine Learning: Understanding of loss functions and optimization, Machine Learning: Ability to train and evaluate classification models using scikit-learn, \"Machine Learning: Ability to interpret common evaluation metrics such as accuracy, precision, recall and ROC AUC\", Machine Learning: Ability to use bootstrapping for uncertainty estimation, Machine Learning: Understanding of data leakage in ML, Machine Learning: Understanding of decision trees and CART models, Machine Learning: Ability to train decision tree models, Machine Learning: Ability to identify and mitigate overfitting, Machine Learning: Understanding of ensemble methods, Machine Learning: Ability to train random forest models, Machine Learning: Ability to train gradient boosted models, Machine Learning: Understanding of model variance and bias, Machine Learning: Understanding of neural network architecture, Machine Learning: Understanding of activation functions, Machine Learning: Understanding of convolutional neural networks, Machine Learning: Ability to build neural networks for image classification, Machine Learning: Ability to interpret saliency maps for model explainability, Machine Learning: Understanding of transfer learning concepts, Machine Learning: Understanding deep learning, Image processing: Understanding of basics of image analysis, Machine Learning: Understanding of potential harms from ML applications, Machine Learning: Awareness of bias in ML models, Machine Learning: Ability to evaluate ML model appropriateness for tasks, Machine Learning: Understanding of dataset quality issues, Machine Learning: Understanding of explainability vs interpretability, Machine Learning: Awareness of adversarial attacks on ML models, Machine Learning: Understanding of dataset shift and model deployment challenges", "Classical statistics: Understanding independent/expanatory variables and dependent/response variables, \"Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous\", Classical statistics: Understanding that variables have distributions, R programming: Proficiency in data manipulation and visualization", "Computing: Ability to use an integrated development environment (IDE) or notebook environment such as Jupyter, \"Computing: Understanding of the difference between scripts, notebooks and interactive shells\", \"Python Programming: Ability to use variables and basic data types such as integers, floats, strings and booleans\", \"Python Programming: Ability to use control flow including if, elif, else statements\", Python Programming: Ability to use loops such as for and while to process data, Python Programming: Ability to define and call functions with arguments and return values, \"Python Programming: Ability to work with built-in data structures such as lists, tuples and dictionaries\", Python Programming: Ability to read from and write to files in Python, Python Programming: Ability to handle basic errors and exceptions in Python code, Python Programming: Ability to write simple Python scripts to automate repetitive tasks, Python Programming: Ability to use virtual environments and install packages using pip", "Classical statistics: Understanding independent/expanatory variables and dependent/response variables, \"Classical statistics: Understanding variable types such as discrete, nominal, ordinal, continuous\", Classical statistics: Understanding that variables have distributions, Python Programming: Ability to work with Python libraries for scientific computing such as NumPy and pandas, Python Programming: Ability to use NumPy arrays for numerical computing, Python Programming: Ability to create and manipulate pandas DataFrames, Python Programming: Ability to import tabular data from CSV and Excel files into pandas, Python Programming: Ability to clean data including handling missing values and inconsistent entries, \"Python Programming: Ability to filter, sort and subset tabular data using pandas\", Python Programming: Ability to group and aggregate data in pandas, Python Programming: Ability to merge and join multiple DataFrames, \"Python Programming: Ability to calculate descriptive statistics such as mean, median, variance and quantiles in Python\", \"Python Programming: Ability to create basic data visualisations including histograms, boxplots, scatter plots and line plots\", Python Programming: Ability to use Matplotlib or similar libraries for publication-quality plots, Data management: Ability to document and share data analysis workflows, Data management: Understanding of data integrity and quality control in bioinformatics, Data management: Ability to use version control systems for data analysis projects", "Network Science: Understanding of graphs as representations of complex systems, \"Network Science: Understanding of nodes, edges, degree and paths in networks\", Network Science: Ability to construct network representations from tabular data, \"Network Science: Ability to calculate basic network measures such as degree distribution, centrality and clustering coefficient\", Network Science: Ability to visualise networks using appropriate layouts, Network Science: Ability to interpret network measures in the context of bioscience or medical case studies", "Machine Learning: Understanding machine learning techniques for bioscience image data , Machine Learning: Understanding of the difference between supervised and unsupervised learning, \"Machine Learning: Understanding of training, validation and test sets\", Machine Learning: Ability to train and evaluate classification models using scikit-learn, Machine Learning: Ability to train and evaluate regression models using scikit-learn, Machine Learning: Ability to apply clustering methods such as k-means, Machine Learning: Ability to use dimensionality reduction techniques such as PCA, Machine Learning: Ability to use cross-validation for model evaluation, \"Machine Learning: Ability to interpret common evaluation metrics such as accuracy, precision, recall and ROC AUC\", Machine Learning: Ability to identify and mitigate overfitting", "\"Artificial Intelligence: Understanding of symbolic, probabilistic and neural approaches to AI\", \"Artificial Intelligence: Understanding of how AI systems are constructed from data, models and objectives\", Artificial Intelligence: Ability to compare strengths and limitations of different AI paradigms for research problems, \"Artificial Intelligence: Awareness of ethical, legal and social implications of AI in bioscience and medicine\", Artificial Intelligence: Understanding of the relationship between classical machine learning and modern AI systems", "Bioscience: Plant phenotyping", "Machine Learning: Understanding machine learning techniques for bioscience image data, Image processing: Understanding of basics of image analysis", "Experimental Design: Understanding foundations of experimental design for machine learning , Machine Learning: Understanding ML methodologies", "Machine Learning: Understanding deep learning, Machine Learning: Use of PyTorch for deep learning", "Artificial Intelligence: Understanding generative AI methods", "Artificial Intelligence: , Machine Learning: Understanding deep learning", "R programming: Proficiency in data manipulation and visualization, R programming: Ability to plot graphs and figures using ggplot2", "Command line: Ability to open a shell such as bash, Command line: Ability to navigate filesystems from the command line, \"Command line: Ability to move, delete, create and copy files and directories from command line\", Command line: Ability to use common command line functions, Command line: Ability to create and run shell scripts", "Artificial Intelligence: Understanding of the relationship between classical machine learning and modern AI systems, Machine Learning: Understanding of neural network architecture, Machine Learning: Understanding of the difference between supervised and unsupervised learning", null, "\"Artificial Intelligence: Awareness of ethical, legal and social implications of AI in bioscience and medicine\", Artificial Intelligence: Ability to compare strengths and limitations of different AI paradigms for research problems, \"Artificial Intelligence: Understanding of symbolic, probabilistic and neural approaches to AI\", \"Artificial Intelligence: Understanding of how AI systems are constructed from data, models and objectives\", Artificial Intelligence: Understanding of the relationship between classical machine learning and modern AI systems", "Data management: Ability to document and share data analysis workflows, Data management: Ability to use version control systems for data analysis projects, Data management: Understanding of data integrity and quality control in bioinformatics", "\"Python Programming: Ability to use variables and basic data types such as integers, floats, strings and booleans\", Python Programming: Ability to use loops such as for and while to process data, \"Python Programming: Ability to use control flow including if, elif, else statements\", Python Programming: Ability to define and call functions with arguments and return values, Python Programming: Ability to use NumPy arrays for numerical computing, Python Programming: Ability to create and manipulate pandas DataFrames, Python Programming: Ability to write simple Python scripts to automate repetitive tasks", "Bioinformatics: Understand the principles of NGS data pre-processing and genome alignment", "R programming: Proficiency in data manipulation and visualization", null, "Classical statistics: Understanding of correlation and multicollinearity", "R programming: Proficiency in data manipulation and visualization", "Bioinformatics: Process and analyse single-cell transcriptomics datasets", "Machine Learning: Use of PyTorch for deep learning, Artificial Intelligence: Adapt modern networks to work on challenging real-world medical imaging data", "Bioinformatics: Process and analyse NGS data using Galaxy", "Bioinformatics: Comprehensive understanding of how to analyse and visualise bulk RNA-seq data", "Bioinformatics: Comprehensive understanding of how to analyse and visualise ATAC-seq data", "\"Bioinformatics: Comprehensive understanding of sequence alignment techniques, focusing on both BLAST and multiple sequence alignment\""], "timeRequired": ["24 hours", "24 hours", "5 hours", "2 hours", "2 hours", "12 hours", "2 hours", "1.5 days", "18 hours", "20 hours", "1 hours", "4 hours", "4 hours", "4 hours", "4 hours", "4 hours", "4 hours", "4 hours", "4 hours", "4 hours", "4 hours", "16.5 hours", "19.5 hours", "16 hours", "12 hours", "4 hours", "12 hours", "12 hours", "6 hours", "6 hours", "6 hours", "8 hours", "8 hours", "4 hours ", "15 hours ", "5 hours", "15 hours ", "15 hours", "30 hours", "20 hours", null, "20 hours", "30 hours", "8 hours", "10 hours", "30 hours", "1.5 hours", "10 hours", "10 hours", "22 hours", "3.5 hours", "7 hours", "30 hours", "5 hours", "2 hours", "2.5 hours", "5 hours"], "url": [null, "https://cloud-span.github.io/nerc-metagenomics-v2q/", "https://cloud-span.github.io/prenomics00-intro", "https://cloud-span.github.io/core-r/02_intro_to_r_and_working_with_data.html", "https://cloud-span.github.io/create-aws-instance-0-overview", "https://cloud-span.github.io/genomics-course", "https://cloud-span.github.io/experimental_design00-overview", "https://healthbioscienceideas.github.io/microscopy-novice/", "https://www.futurelearn.com/courses/introduction-to-image-analysis-for-plant-phenotyping", "https://learninghub.kingshealthpartners.org/course/introduction-to-python-for-health-research", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-fair-clinical-data", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-ena-sequence-submission", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-fair-data-management-solutions", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-rembi-metadata", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-single-cell", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-bioimage-metadata-course", "https://fellowship.elixiruknode.org/latest/carpentries-course-rna-seq-ml", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-fair-galaxy-training", "https://fellowship.elixiruknode.org/latest/gtn-tutorial-fair-in-a-nutshell", "https://fellowship.elixiruknode.org/latest/carpentries-course-agricultural-data-long-term", "https://fellowship.elixiruknode.org/latest/carpentries-course-fair-pointers", "https://carpentries-incubator.github.io/workflows-nextflow/", "https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/", "https://carpentries-incubator.github.io/fair-bio-practice/", "https://carpentries-incubator.github.io/high-dimensional-stats-r/", "https://carpentries-incubator.github.io/introduction-to-conda-for-data-scientists/", "https://carpentries-incubator.github.io/machine-learning-novice-python/", "https://carpentries-incubator.github.io/statistics-r-public-health/", "https://learntodiscover.ai/product/basic-python-2025/", "https://learntodiscover.ai/product/__trashed/", "https://learntodiscover.ai/product/network-science-module/", "https://learntodiscover.ai/product/machine-learning/", "https://learntodiscover.ai/product/artificial-intelligence-ai/", "https://www.futurelearn.com/courses/plant-phenotyping-technologies", "https://www.futurelearn.com/courses/machine-learning-for-image-data", "https://www.futurelearn.com/courses/experimental-design-for-machine-learning", "https://www.futurelearn.com/courses/deep-learning-for-bioscientists", "URL Pending ", "https://learninghub.kingshealthpartners.org/course/ai-in-health-analytics", "https://learninghub.kingshealthpartners.org/course/introduction-to-r-for-health-research", "https://learninghub.kingshealthpartners.org/course/unix", "https://learninghub.kingshealthpartners.org/course/machine-learning-for-health-research", "https://learninghub.kingshealthpartners.org/course/prediction-modelling-for-health-research", "https://learninghub.kingshealthpartners.org/course/demystifying-ai", "https://learninghub.kingshealthpartners.org/course/using-spreadsheets", "https://learninghub.kingshealthpartners.org/course/intermediate-python-and-software-engineering", "https://learninghub.kingshealthpartners.org/course/ngs-introduction-and-pre-processing-pipeline", "https://learninghub.kingshealthpartners.org/course/basic-r-with-data-carpentry", "https://learninghub.kingshealthpartners.org/course/natural-language-processing-for-health-research", "https://learninghub.kingshealthpartners.org/course/introduction-to-statistics", "https://learninghub.kingshealthpartners.org/course/statistics-with-r", "https://learninghub.kingshealthpartners.org/course/ngs-single-cell-rna-sequencing-analysis", "https://learninghub.kingshealthpartners.org/course/applied-artificial-intelligence-for-health-research", "https://learninghub.kingshealthpartners.org/course/ngs-on-the-galaxy-platform", "https://learninghub.kingshealthpartners.org/course/next-generation-sequencing-rna-seq", "https://learninghub.kingshealthpartners.org/course/ngs-atac-seq", "https://learninghub.kingshealthpartners.org/course/sequence-alignment-blast"], "version": [null, "1.2", "0.1.0", null, null, null, null, "0.9", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, "2025.1", "2025.1", "2025.1", "2025.1", "2025.1", null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null]}}]}</script>
</div>
</div>
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb2" data-startfrom="26" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 25;"><span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co">// This cell transposes the data presented from Python, as numpy &amp; OJS expect data in opposite order</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>course_data <span class="op">=</span> <span class="fu">transpose</span>(course_data_cached)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-1" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb3" data-startfrom="35" data-source-offset="-270"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 34;"><span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="kw">function</span> <span class="fu">csv_column_set</span>(ds<span class="op">,</span> col<span class="op">,</span> inject_null) {</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> all_values <span class="op">=</span> ds<span class="op">.</span><span class="fu">flatMap</span>(d <span class="kw">=&gt;</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>    d[col]</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>      <span class="op">?</span> d[col]<span class="op">.</span><span class="fu">split</span>(<span class="st">","</span>)<span class="op">.</span><span class="fu">map</span>(s <span class="kw">=&gt;</span> s<span class="op">.</span><span class="fu">trim</span>())</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>      <span class="op">:</span> []</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  )<span class="op">;</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  all_values <span class="op">=</span> <span class="bu">Array</span><span class="op">.</span><span class="fu">from</span>(<span class="kw">new</span> <span class="bu">Set</span>(all_values))<span class="op">.</span><span class="fu">sort</span>()</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(inject_null) {</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>    all_values<span class="op">.</span><span class="fu">unshift</span>(<span class="kw">null</span>)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  <span class="cf">return</span> all_values</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>audiences <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"audience"</span><span class="op">,</span> <span class="kw">true</span>)<span class="op">;</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>access_modes <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"accessMode"</span><span class="op">,</span> <span class="kw">false</span>)<span class="op">;</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>education_levels <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"educationalLevel"</span><span class="op">,</span> <span class="kw">false</span>)<span class="op">;</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>resource_types <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"learningResourceType"</span><span class="op">,</span> <span class="kw">true</span>)<span class="op">;</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>providers <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"provider"</span><span class="op">,</span> <span class="kw">true</span>)<span class="op">;</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>times <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"timeRequired"</span><span class="op">,</span> <span class="kw">true</span>)<span class="op">;</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>keywords <span class="op">=</span> <span class="fu">csv_column_set</span>(course_data<span class="op">,</span> <span class="st">"keywords"</span><span class="op">,</span> <span class="kw">false</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-1" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-2" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-3" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-4" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-5" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-6" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-7" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-2-8" data-nodetype="declaration">

</div>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb4" data-startfrom="56" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 55;"><span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a><span class="co">// Trim the search query</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>search_query <span class="op">=</span> (search_terms <span class="op">??</span> <span class="st">""</span>)<span class="op">.</span><span class="fu">toLowerCase</span>()<span class="op">.</span><span class="fu">trim</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-3" data-nodetype="declaration">

</div>
</div>
</div>
<div class="primary-search">
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb5" data-startfrom="62" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 61;"><span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a><span class="co">// The main search box.</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>viewof search_terms <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">text</span>({<span class="dt">placeholder</span><span class="op">:</span> <span class="st">"Search Training Resources..."</span>})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-4" data-nodetype="declaration">

</div>
</div>
</div>
<div id="search-filters" style="display: none;">
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb6" data-startfrom="72" data-source-offset="-229"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 71;"><span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>viewof selected_audience <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">select</span>(audiences<span class="op">,</span> {<span class="dt">label</span><span class="op">:</span> <span class="st">"Audience"</span><span class="op">,</span> <span class="dt">value</span><span class="op">:</span> <span class="kw">null</span>})</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>viewof selected_educational_level <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">checkbox</span>(education_levels<span class="op">,</span> {<span class="dt">label</span><span class="op">:</span> <span class="st">"Level"</span><span class="op">,</span> <span class="dt">value</span><span class="op">:</span> education_levels})</span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>viewof selected_resource_type <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">select</span>(resource_types<span class="op">,</span> {<span class="dt">label</span><span class="op">:</span> <span class="st">"Type"</span><span class="op">,</span> <span class="dt">value</span><span class="op">:</span> <span class="kw">null</span>})</span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>viewof selected_time <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">select</span>(times<span class="op">,</span> {<span class="dt">label</span><span class="op">:</span> <span class="st">"Time Required"</span><span class="op">,</span> <span class="dt">value</span><span class="op">:</span> <span class="kw">null</span>})</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>viewof selected_provider <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">select</span>(providers<span class="op">,</span> {<span class="dt">label</span><span class="op">:</span> <span class="st">"Provider"</span><span class="op">,</span> <span class="dt">value</span><span class="op">:</span> <span class="kw">null</span>})</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a>viewof reset_filters <span class="op">=</span> Inputs<span class="op">.</span><span class="fu">button</span>(<span class="st">"reset filters"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-5-1" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-5-2" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-5-3" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-5-4" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-5-5" data-nodetype="declaration">

</div>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<div id="ojs-cell-5-6" data-nodetype="declaration">

</div>
</div>
</div>
</div>
</div>
<div class="search-filter-toggle">
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb7" data-startfrom="82" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 81;"><span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a><span class="co">// This cell implements a show/hide button for extra filters.</span></span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a><span class="co">// This works via a JavaScript function that finds and toggles the div display style.</span></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>  <span class="kw">const</span> toggleLink <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">"a"</span>)<span class="op">;</span></span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>  toggleLink<span class="op">.</span><span class="at">href</span> <span class="op">=</span> <span class="st">"#"</span><span class="op">;</span></span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>  toggleLink<span class="op">.</span><span class="at">textContent</span> <span class="op">=</span> <span class="st">"show filters"</span><span class="op">;</span></span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>  toggleLink<span class="op">.</span><span class="fu">addEventListener</span>(<span class="st">"click"</span><span class="op">,</span> (<span class="bu">event</span>) <span class="kw">=&gt;</span> {</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>    <span class="bu">event</span><span class="op">.</span><span class="fu">preventDefault</span>()<span class="op">;</span></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>    <span class="kw">const</span> filter_div <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(<span class="st">"search-filters"</span>)<span class="op">;</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (filter_div<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">===</span> <span class="st">"none"</span>) {</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a>      toggleLink<span class="op">.</span><span class="at">textContent</span> <span class="op">=</span> <span class="st">"hide filters"</span><span class="op">;</span></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>      filter_div<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">=</span> <span class="st">"block"</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a>      toggleLink<span class="op">.</span><span class="at">textContent</span> <span class="op">=</span> <span class="st">"show filters"</span><span class="op">;</span></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a>      filter_div<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">=</span> <span class="st">"none"</span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>  })<span class="op">;</span></span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>  <span class="cf">return</span> toggleLink<span class="op">;</span></span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-6" data-nodetype="expression">

</div>
</div>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb8" data-startfrom="107" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 106;"><span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a><span class="co">// Performs filtering of the complete (`course_data`) data into the filtered (`course_filtered`) set.</span></span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a><span class="co">// If there is no search, and nothing has been selected in the filters then we return everything.</span></span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>course_filtered <span class="op">=</span></span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>  (</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>    (<span class="op">!</span>selected_audience) <span class="op">&amp;&amp;</span></span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>    (selected_educational_level <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">&amp;&amp;</span></span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>    (selected_resource_type <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">&amp;&amp;</span></span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a>    (selected_time <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">&amp;&amp;</span></span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>    (selected_provider <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">&amp;&amp;</span></span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a>    (<span class="op">!</span>search_query)</span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>  <span class="op">?</span> course_data<span class="op">.</span><span class="fu">filter</span>(row <span class="kw">=&gt;</span> row)</span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>  <span class="op">:</span> course_data<span class="op">.</span><span class="fu">filter</span>(row <span class="kw">=&gt;</span></span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>      <span class="co">// audience</span></span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a>      (<span class="op">!</span>selected_audience <span class="op">||</span> <span class="op">!</span>row<span class="op">.</span><span class="at">audience</span> <span class="op">||</span> row<span class="op">.</span><span class="at">audience</span><span class="op">.</span><span class="fu">includes</span>(selected_audience)) <span class="op">&amp;&amp;</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a>      <span class="co">// educational level</span></span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a>      ((selected_educational_level <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">||</span> (selected_educational_level <span class="op">??</span> [])<span class="op">.</span><span class="fu">includes</span>(row<span class="op">.</span><span class="at">educationalLevel</span>)) <span class="op">&amp;&amp;</span></span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>      <span class="co">// resource type</span></span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a>      ((selected_resource_type <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">||</span> (selected_resource_type <span class="op">??</span> [])<span class="op">.</span><span class="fu">includes</span>(row<span class="op">.</span><span class="at">learningResourceType</span>)) <span class="op">&amp;&amp;</span></span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a>      <span class="co">// time</span></span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a>      ((selected_time <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">||</span> (selected_time <span class="op">??</span> [])<span class="op">.</span><span class="fu">includes</span>(row<span class="op">.</span><span class="at">timeRequired</span>)) <span class="op">&amp;&amp;</span></span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a>      <span class="co">// provider</span></span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a>      ((selected_provider <span class="op">??</span> [])<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span> <span class="op">||</span> (selected_provider <span class="op">??</span> [])<span class="op">.</span><span class="fu">includes</span>(row<span class="op">.</span><span class="at">provider</span>)) <span class="op">&amp;&amp;</span></span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>      <span class="co">// search</span></span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a>      (<span class="op">!</span>search_query <span class="op">||</span></span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a>        row<span class="op">.</span><span class="at">headline</span><span class="op">.</span><span class="fu">toLowerCase</span>()<span class="op">.</span><span class="fu">includes</span>(search_query) <span class="op">||</span></span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a>        (row<span class="op">.</span><span class="at">description</span>  <span class="op">&amp;&amp;</span> row<span class="op">.</span><span class="at">description</span><span class="op">.</span><span class="fu">includes</span>(search_query)) <span class="op">||</span></span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a>        (row<span class="op">.</span><span class="at">projectFunding</span>  <span class="op">&amp;&amp;</span> row<span class="op">.</span><span class="at">projectFunding</span><span class="op">.</span><span class="fu">includes</span>(search_query)) <span class="op">||</span></span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>        (row<span class="op">.</span><span class="at">identifier</span>  <span class="op">&amp;&amp;</span> row<span class="op">.</span><span class="at">identifier</span><span class="op">.</span><span class="fu">includes</span>(search_query))</span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a>    )</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-7" data-nodetype="declaration">

</div>
</div>
</div>
<div id="primary-search-results">
<div class="cell">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb9" data-startfrom="143" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 142;"><span id="cb9-143"><a href="#cb9-143" aria-hidden="true" tabindex="-1"></a><span class="co">// This cell iterates through the the filtered course results, and builds the displayed output.</span></span>
<span id="cb9-144"><a href="#cb9-144" aria-hidden="true" tabindex="-1"></a><span class="co">// If there are no selected results, we show a placeholder &lt;div&gt;.</span></span>
<span id="cb9-145"><a href="#cb9-145" aria-hidden="true" tabindex="-1"></a><span class="co">// All filtered courses are displayed, sorted first by `featured` status then in the CSV file order.</span></span>
<span id="cb9-146"><a href="#cb9-146" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb9-147"><a href="#cb9-147" aria-hidden="true" tabindex="-1"></a>  <span class="kw">const</span> search_results <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(<span class="st">"primary-search-results"</span>)<span class="op">;</span></span>
<span id="cb9-148"><a href="#cb9-148" aria-hidden="true" tabindex="-1"></a>  <span class="kw">const</span> empty_results <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(<span class="st">"empty-primary-search-results"</span>)<span class="op">;</span></span>
<span id="cb9-149"><a href="#cb9-149" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (course_filtered<span class="op">.</span><span class="at">length</span> <span class="op">===</span> <span class="dv">0</span>) {</span>
<span id="cb9-150"><a href="#cb9-150" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> <span class="fu">html</span><span class="vs">`</span></span>
<span id="cb9-151"><a href="#cb9-151" aria-hidden="true" tabindex="-1"></a><span class="vs">      &lt;div class="course-data-empty"&gt;</span></span>
<span id="cb9-152"><a href="#cb9-152" aria-hidden="true" tabindex="-1"></a><span class="vs">        &lt;p&gt;No results to display&lt;/p&gt;</span></span>
<span id="cb9-153"><a href="#cb9-153" aria-hidden="true" tabindex="-1"></a><span class="vs">      &lt;/div&gt;</span></span>
<span id="cb9-154"><a href="#cb9-154" aria-hidden="true" tabindex="-1"></a><span class="vs">    `</span></span>
<span id="cb9-155"><a href="#cb9-155" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb9-156"><a href="#cb9-156" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> <span class="fu">html</span><span class="vs">`</span></span>
<span id="cb9-157"><a href="#cb9-157" aria-hidden="true" tabindex="-1"></a><span class="vs">      </span><span class="sc">${</span>course_filtered<span class="op">.</span><span class="fu">slice</span>()<span class="op">.</span><span class="fu">sort</span>((a<span class="op">,</span> b) <span class="kw">=&gt;</span> {</span>
<span id="cb9-158"><a href="#cb9-158" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(a<span class="op">.</span><span class="at">featured</span> <span class="op">===</span> <span class="st">"yes"</span> <span class="op">&amp;&amp;</span> b<span class="op">.</span><span class="at">featured</span> <span class="op">===</span> <span class="st">"no"</span>) <span class="cf">return</span> <span class="op">-</span><span class="dv">1</span><span class="op">;</span></span>
<span id="cb9-159"><a href="#cb9-159" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(a<span class="op">.</span><span class="at">featured</span> <span class="op">===</span> <span class="st">"no"</span> <span class="op">&amp;&amp;</span> b<span class="op">.</span><span class="at">featured</span> <span class="op">===</span> <span class="st">"yes"</span>) <span class="cf">return</span> <span class="dv">1</span><span class="op">;</span></span>
<span id="cb9-160"><a href="#cb9-160" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="dv">0</span><span class="op">;</span></span>
<span id="cb9-161"><a href="#cb9-161" aria-hidden="true" tabindex="-1"></a>      })<span class="op">.</span><span class="fu">map</span>((row<span class="op">,</span> row_index) <span class="kw">=&gt;</span> <span class="fu">html</span> <span class="vs">`</span></span>
<span id="cb9-162"><a href="#cb9-162" aria-hidden="true" tabindex="-1"></a><span class="vs">        &lt;div class="</span><span class="sc">${</span>row<span class="op">.</span><span class="at">featured</span> <span class="op">===</span> <span class="st">"yes"</span> <span class="op">?</span> <span class="st">"course-data featured"</span> <span class="op">:</span> <span class="st">"course-data"</span><span class="sc">}</span><span class="vs">" id="course_data_</span><span class="sc">${</span>row_index<span class="sc">}</span><span class="vs">"&gt;</span></span>
<span id="cb9-163"><a href="#cb9-163" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;h3&gt;</span></span>
<span id="cb9-164"><a href="#cb9-164" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;a href=</span><span class="sc">${</span>row<span class="op">.</span><span class="at">url</span><span class="sc">}</span><span class="vs">&gt;</span><span class="sc">${</span>row<span class="op">.</span><span class="at">headline</span><span class="sc">}</span><span class="vs">&lt;/a&gt;</span></span>
<span id="cb9-165"><a href="#cb9-165" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;/h3&gt;</span></span>
<span id="cb9-166"><a href="#cb9-166" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;p&gt;</span><span class="sc">${</span>row<span class="op">.</span><span class="at">description</span><span class="sc">}</span><span class="vs">&lt;/p&gt;</span></span>
<span id="cb9-167"><a href="#cb9-167" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;input type="checkbox" class="toggle-state" id="toggle-state-</span><span class="sc">${</span>row_index<span class="sc">}</span><span class="vs">"&gt;</span></span>
<span id="cb9-168"><a href="#cb9-168" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;label for="toggle-state-</span><span class="sc">${</span>row_index<span class="sc">}</span><span class="vs">" class="toggle-button"&gt;&lt;/label&gt;</span></span>
<span id="cb9-169"><a href="#cb9-169" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;div class="course-data-extras" id="course_data_extras_</span><span class="sc">${</span>row_index<span class="sc">}</span><span class="vs">"&gt;</span></span>
<span id="cb9-170"><a href="#cb9-170" aria-hidden="true" tabindex="-1"></a><span class="vs">            &lt;table&gt;</span></span>
<span id="cb9-171"><a href="#cb9-171" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;tr&gt;</span></span>
<span id="cb9-172"><a href="#cb9-172" aria-hidden="true" tabindex="-1"></a><span class="vs">                &lt;th scope="row"&gt;Pre-requisites&lt;/th&gt;</span></span>
<span id="cb9-173"><a href="#cb9-173" aria-hidden="true" tabindex="-1"></a><span class="vs">                &lt;td&gt;</span><span class="sc">${</span>row<span class="op">.</span><span class="at">competencyRequired</span><span class="sc">}</span><span class="vs">&lt;/td&gt;</span></span>
<span id="cb9-174"><a href="#cb9-174" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;/tr&gt;</span></span>
<span id="cb9-175"><a href="#cb9-175" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;tr&gt;</span></span>
<span id="cb9-176"><a href="#cb9-176" aria-hidden="true" tabindex="-1"></a><span class="vs">                &lt;th scope="row"&gt;Teaches&lt;/th&gt;</span></span>
<span id="cb9-177"><a href="#cb9-177" aria-hidden="true" tabindex="-1"></a><span class="vs">                &lt;td&gt;</span><span class="sc">${</span>row<span class="op">.</span><span class="at">teaches</span><span class="sc">}</span><span class="vs">&lt;/td&gt;</span></span>
<span id="cb9-178"><a href="#cb9-178" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;/tr&gt;</span></span>
<span id="cb9-179"><a href="#cb9-179" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;tr&gt;</span></span>
<span id="cb9-180"><a href="#cb9-180" aria-hidden="true" tabindex="-1"></a><span class="vs">                &lt;th scope="row"&gt;Provider&lt;/th&gt;</span></span>
<span id="cb9-181"><a href="#cb9-181" aria-hidden="true" tabindex="-1"></a><span class="vs">                &lt;td&gt;</span><span class="sc">${</span>row<span class="op">.</span><span class="at">projectFunding</span><span class="sc">}</span><span class="vs">&lt;/td&gt;</span></span>
<span id="cb9-182"><a href="#cb9-182" aria-hidden="true" tabindex="-1"></a><span class="vs">              &lt;/tr&gt;</span></span>
<span id="cb9-183"><a href="#cb9-183" aria-hidden="true" tabindex="-1"></a><span class="vs">            &lt;/table&gt;</span></span>
<span id="cb9-184"><a href="#cb9-184" aria-hidden="true" tabindex="-1"></a><span class="vs">          &lt;/div&gt;</span></span>
<span id="cb9-185"><a href="#cb9-185" aria-hidden="true" tabindex="-1"></a><span class="vs">        &lt;/div&gt;</span></span>
<span id="cb9-186"><a href="#cb9-186" aria-hidden="true" tabindex="-1"></a><span class="vs">    `</span>)<span class="sc">}</span><span class="vs">`</span></span>
<span id="cb9-187"><a href="#cb9-187" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-188"><a href="#cb9-188" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-8" data-nodetype="expression">

</div>
</div>
</div>
</div>
<div class="cell hidden">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb10" data-startfrom="194" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 193;"><span id="cb10-194"><a href="#cb10-194" aria-hidden="true" tabindex="-1"></a><span class="co">// This cell fires whenever the "reset filters" button is pressed.</span></span>
<span id="cb10-195"><a href="#cb10-195" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb10-196"><a href="#cb10-196" aria-hidden="true" tabindex="-1"></a>  reset_filters<span class="op">;</span> <span class="co">// This updates whenever the reset button fires.</span></span>
<span id="cb10-197"><a href="#cb10-197" aria-hidden="true" tabindex="-1"></a>  <span class="kw">function</span> <span class="fu">resetInput</span>(view<span class="op">,</span> value) {</span>
<span id="cb10-198"><a href="#cb10-198" aria-hidden="true" tabindex="-1"></a>    <span class="kw">const</span> el <span class="op">=</span> view<span class="op">.</span><span class="fu">querySelector</span>(<span class="st">"select, input"</span>)<span class="op">;</span></span>
<span id="cb10-199"><a href="#cb10-199" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="op">!</span>el) <span class="cf">return</span><span class="op">;</span></span>
<span id="cb10-200"><a href="#cb10-200" aria-hidden="true" tabindex="-1"></a>    el<span class="op">.</span><span class="at">value</span> <span class="op">=</span> value<span class="op">;</span></span>
<span id="cb10-201"><a href="#cb10-201" aria-hidden="true" tabindex="-1"></a>    view<span class="op">.</span><span class="at">value</span> <span class="op">=</span> value<span class="op">;</span></span>
<span id="cb10-202"><a href="#cb10-202" aria-hidden="true" tabindex="-1"></a>    el<span class="op">.</span><span class="fu">dispatchEvent</span>(<span class="kw">new</span> <span class="bu">Event</span>(<span class="st">"input"</span><span class="op">,</span> { <span class="dt">bubbles</span><span class="op">:</span> <span class="kw">true</span> }))<span class="op">;</span></span>
<span id="cb10-203"><a href="#cb10-203" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-204"><a href="#cb10-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">resetInput</span>(viewof selected_audience<span class="op">,</span> <span class="kw">null</span>)<span class="op">;</span></span>
<span id="cb10-205"><a href="#cb10-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">resetInput</span>(viewof selected_educational_level<span class="op">,</span> [])<span class="op">;</span></span>
<span id="cb10-206"><a href="#cb10-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">resetInput</span>(viewof selected_resource_type<span class="op">,</span> <span class="kw">null</span>)<span class="op">;</span></span>
<span id="cb10-207"><a href="#cb10-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">resetInput</span>(viewof selected_time<span class="op">,</span> <span class="kw">null</span>)<span class="op">;</span></span>
<span id="cb10-208"><a href="#cb10-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">resetInput</span>(viewof selected_provider<span class="op">,</span> <span class="kw">null</span>)<span class="op">;</span></span>
<span id="cb10-209"><a href="#cb10-209" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display hidden">
<div id="ojs-cell-9" data-nodetype="expression">

</div>
</div>
</div>
<div class="cell" data-eval="false">
<details class="code-fold hidden">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code hidden" id="cb11" data-startfrom="214" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 213;"><span id="cb11-214"><a href="#cb11-214" aria-hidden="true" tabindex="-1"></a><span class="co">// This cell displays a table of all `course_data` for debugging only.</span></span>
<span id="cb11-215"><a href="#cb11-215" aria-hidden="true" tabindex="-1"></a>Inputs<span class="op">.</span><span class="fu">table</span>(course_data)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display">
<div id="ojs-cell-10" data-nodetype="expression">

</div>
</div>
</div>



</main> <!-- /main -->
<script type="ojs-module-contents">
{"contents":[{"methodName":"interpret","cellName":"ojs-cell-1","inline":false,"source":"// This cell transposes the data presented from Python, as numpy & OJS expect data in opposite order\ncourse_data = transpose(course_data_cached)\n"},{"methodName":"interpret","cellName":"ojs-cell-2","inline":false,"source":"// This cell defines the possible values we can use to populate the filter dropdowns.#\n// We do this by pulling out all options for specific columns from the course_data object, splitting on commas, and sorting.\n// The `csv_column_set` function simplifies this process.\nfunction csv_column_set(ds, col, inject_null) {\n  let all_values = ds.flatMap(d =>\n    d[col]\n      ? d[col].split(\",\").map(s => s.trim())\n      : []\n  );\n  all_values = Array.from(new Set(all_values)).sort()\n  if(inject_null) {\n    all_values.unshift(null)\n  }\n  return all_values\n}\naudiences = csv_column_set(course_data, \"audience\", true);\naccess_modes = csv_column_set(course_data, \"accessMode\", false);\neducation_levels = csv_column_set(course_data, \"educationalLevel\", false);\nresource_types = csv_column_set(course_data, \"learningResourceType\", true);\nproviders = csv_column_set(course_data, \"provider\", true);\ntimes = csv_column_set(course_data, \"timeRequired\", true);\nkeywords = csv_column_set(course_data, \"keywords\", false);\n"},{"methodName":"interpret","cellName":"ojs-cell-3","inline":false,"source":"// Trim the search query\nsearch_query = (search_terms ?? \"\").toLowerCase().trim()\n"},{"methodName":"interpret","cellName":"ojs-cell-4","inline":false,"source":"// The main search box.\nviewof search_terms = Inputs.text({placeholder: \"Search Training Resources...\"})\n"},{"methodName":"interpret","cellName":"ojs-cell-5","inline":false,"source":"// This cell holds all of the search filters that can refine the currently found courses.\n// The reset button allows a quick way to re-set the default values.\n// The action cell is at the bottom and does not generate any output.\nviewof selected_audience = Inputs.select(audiences, {label: \"Audience\", value: null})\nviewof selected_educational_level = Inputs.checkbox(education_levels, {label: \"Level\", value: education_levels})\nviewof selected_resource_type = Inputs.select(resource_types, {label: \"Type\", value: null})\nviewof selected_time = Inputs.select(times, {label: \"Time Required\", value: null})\nviewof selected_provider = Inputs.select(providers, {label: \"Provider\", value: null})\nviewof reset_filters = Inputs.button(\"reset filters\")\n"},{"methodName":"interpret","cellName":"ojs-cell-6","inline":false,"source":"// This cell implements a show/hide button for extra filters.\n// This works via a JavaScript function that finds and toggles the div display style.\n{\n  const toggleLink = document.createElement(\"a\");\n  toggleLink.href = \"#\";\n  toggleLink.textContent = \"show filters\";\n  toggleLink.addEventListener(\"click\", (event) => {\n    event.preventDefault();\n    const filter_div = document.getElementById(\"search-filters\");\n    if (filter_div.style.display === \"none\") {\n      toggleLink.textContent = \"hide filters\";\n      filter_div.style.display = \"block\"\n    } else {\n      toggleLink.textContent = \"show filters\";\n      filter_div.style.display = \"none\"\n    }\n  });\n  return toggleLink;\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-7","inline":false,"source":"// Performs filtering of the complete (`course_data`) data into the filtered (`course_filtered`) set.\n// If there is no search, and nothing has been selected in the filters then we return everything.\ncourse_filtered =\n  (\n    (!selected_audience) &&\n    (selected_educational_level ?? []).length === 0 &&\n    (selected_resource_type ?? []).length === 0 &&\n    (selected_time ?? []).length === 0 &&\n    (selected_provider ?? []).length === 0 &&\n    (!search_query)\n  )\n  ? course_data.filter(row => row)\n  : course_data.filter(row =>\n      // audience\n      (!selected_audience || !row.audience || row.audience.includes(selected_audience)) &&\n      // educational level\n      ((selected_educational_level ?? []).length === 0 || (selected_educational_level ?? []).includes(row.educationalLevel)) &&\n      // resource type\n      ((selected_resource_type ?? []).length === 0 || (selected_resource_type ?? []).includes(row.learningResourceType)) &&\n      // time\n      ((selected_time ?? []).length === 0 || (selected_time ?? []).includes(row.timeRequired)) &&\n      // provider\n      ((selected_provider ?? []).length === 0 || (selected_provider ?? []).includes(row.provider)) &&\n      // search\n      (!search_query ||\n        row.headline.toLowerCase().includes(search_query) ||\n        (row.description  && row.description.includes(search_query)) ||\n        (row.projectFunding  && row.projectFunding.includes(search_query)) ||\n        (row.identifier  && row.identifier.includes(search_query))\n      )\n    )\n"},{"methodName":"interpret","cellName":"ojs-cell-8","inline":false,"source":"// This cell iterates through the the filtered course results, and builds the displayed output.\n// If there are no selected results, we show a placeholder <div>.\n// All filtered courses are displayed, sorted first by `featured` status then in the CSV file order.\n{\n  const search_results = document.getElementById(\"primary-search-results\");\n  const empty_results = document.getElementById(\"empty-primary-search-results\");\n  if (course_filtered.length === 0) {\n    return html`\n      <div class=\"course-data-empty\">\n        <p>No results to display</p>\n      </div>\n    `\n  } else {\n    return html`\n      ${course_filtered.slice().sort((a, b) => {\n        if(a.featured === \"yes\" && b.featured === \"no\") return -1;\n        if(a.featured === \"no\" && b.featured === \"yes\") return 1;\n        return 0;\n      }).map((row, row_index) => html `\n        <div class=\"${row.featured === \"yes\" ? \"course-data featured\" : \"course-data\"}\" id=\"course_data_${row_index}\">\n          <h3>\n              <a href=${row.url}>${row.headline}</a>\n          </h3>\n          <p>${row.description}</p>\n          <input type=\"checkbox\" class=\"toggle-state\" id=\"toggle-state-${row_index}\">\n          <label for=\"toggle-state-${row_index}\" class=\"toggle-button\"></label>\n          <div class=\"course-data-extras\" id=\"course_data_extras_${row_index}\">\n            <table>\n              <tr>\n                <th scope=\"row\">Pre-requisites</th>\n                <td>${row.competencyRequired}</td>\n              </tr>\n              <tr>\n                <th scope=\"row\">Teaches</th>\n                <td>${row.teaches}</td>\n              </tr>\n              <tr>\n                <th scope=\"row\">Provider</th>\n                <td>${row.projectFunding}</td>\n              </tr>\n            </table>\n          </div>\n        </div>\n    `)}`\n  }\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-9","inline":false,"source":"// This cell fires whenever the \"reset filters\" button is pressed.\n{\n  reset_filters; // This updates whenever the reset button fires.\n  function resetInput(view, value) {\n    const el = view.querySelector(\"select, input\");\n    if (!el) return;\n    el.value = value;\n    view.value = value;\n    el.dispatchEvent(new Event(\"input\", { bubbles: true }));\n  }\n  resetInput(viewof selected_audience, null);\n  resetInput(viewof selected_educational_level, []);\n  resetInput(viewof selected_resource_type, null);\n  resetInput(viewof selected_time, null);\n  resetInput(viewof selected_provider, null);\n}\n"},{"methodName":"interpretQuiet","source":"shinyInput('search_terms')"},{"methodName":"interpretQuiet","source":"shinyInput('selected_audience')"},{"methodName":"interpretQuiet","source":"shinyInput('selected_educational_level')"},{"methodName":"interpretQuiet","source":"shinyInput('selected_resource_type')"},{"methodName":"interpretQuiet","source":"shinyInput('selected_time')"},{"methodName":"interpretQuiet","source":"shinyInput('selected_provider')"},{"methodName":"interpretQuiet","source":"shinyInput('reset_filters')"}]}
</script>
<script type="module">
if (window.location.protocol === "file:") { alert("The OJS runtime does not work with file:// URLs. Please use a web server to view this document."); }
window._ojs.paths.runtimeToDoc = "../..";
window._ojs.paths.runtimeToRoot = "../..";
window._ojs.paths.docToRoot = "";
window._ojs.selfContained = false;
window._ojs.runtime.interpretFromScriptTags();
</script>
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/UKRI-Digital-Research-Skills-Catalyst\.github\.io\/digital-skills-website\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      <ul class="footer-items list-unstyled">
    <li class="nav-item">
    <a class="nav-link" href="https://fusion-cdt.ac.uk/about/contact/">
<p>contact</p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://www.york.ac.uk/about/legal-statements/accessibility/">
<p>accessibility</p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://www.york.ac.uk/about/legal-statements/#privacy">
<p>privacy</p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="https://www.york.ac.uk/about/legal-statements/cookies/">
<p>cookies</p>
</a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="./#">
<p>funding</p>
</a>
  </li>  
</ul>
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://uk.linkedin.com/company/ukri-digital-research-skills-catalyst">
      <i class="bi bi-linkedin" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://bsky.app/profile/ukriskillscatalyst.bsky.social">
      <i class="bi bi-bluesky" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>